<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>S.5 Discussion and Limitations | SPARTYN Supplementary material</title>
  <meta name="description" content="Supplementary material related to the SPARTYN pipeline" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="S.5 Discussion and Limitations | SPARTYN Supplementary material" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Supplementary material related to the SPARTYN pipeline" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="S.5 Discussion and Limitations | SPARTYN Supplementary material" />
  
  <meta name="twitter:description" content="Supplementary material related to the SPARTYN pipeline" />
  

<meta name="author" content="Nate Osher" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="results.html"/>

<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="materials-and-methods.html"><a href="materials-and-methods.html"><i class="fa fa-check"></i><b>3</b> Materials and Methods</a>
<ul>
<li class="chapter" data-level="3.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#overview"><i class="fa fa-check"></i><b>3.1</b> Overview</a></li>
<li class="chapter" data-level="3.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#spatial-point-process-models"><i class="fa fa-check"></i><b>3.2</b> Spatial Point Process Models</a></li>
<li class="chapter" data-level="3.3" data-path="materials-and-methods.html"><a href="materials-and-methods.html#pseudolikelihood-function"><i class="fa fa-check"></i><b>3.3</b> Pseudolikelihood Function</a></li>
<li class="chapter" data-level="3.4" data-path="materials-and-methods.html"><a href="materials-and-methods.html#intensity-thresholding-and-partition"><i class="fa fa-check"></i><b>3.4</b> Intensity Thresholding and Partition</a></li>
<li class="chapter" data-level="3.5" data-path="materials-and-methods.html"><a href="materials-and-methods.html#cell-type-interaction-probability"><i class="fa fa-check"></i><b>3.5</b> Cell Type Interaction Probability</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="results.html"><a href="results.html"><i class="fa fa-check"></i><b>4</b> Results</a>
<ul>
<li class="chapter" data-level="4.1" data-path="results.html"><a href="results.html#simulation"><i class="fa fa-check"></i><b>4.1</b> Simulation</a></li>
<li class="chapter" data-level="4.2" data-path="results.html"><a href="results.html#application"><i class="fa fa-check"></i><b>4.2</b> Application</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="results.html"><a href="results.html#genomic-associations"><i class="fa fa-check"></i><b>4.2.1</b> Genomic Associations</a></li>
<li class="chapter" data-level="4.2.2" data-path="results.html"><a href="results.html#association-with-deconvolution-data"><i class="fa fa-check"></i><b>4.2.2</b> Association with Deconvolution Data</a></li>
<li class="chapter" data-level="4.2.3" data-path="results.html"><a href="results.html#association-with-other-outcomes-of-interest"><i class="fa fa-check"></i><b>4.2.3</b> Association with Other Outcomes of Interest</a></li>
<li class="chapter" data-level="4.2.4" data-path="results.html"><a href="results.html#survival-analysis"><i class="fa fa-check"></i><b>4.2.4</b> Survival Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="discussion-and-limitations.html"><a href="discussion-and-limitations.html"><i class="fa fa-check"></i><b>5</b> Discussion and Limitations</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">SPARTYN Supplementary material</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="discussion-and-limitations" class="section level1" number="5">
<h1><span class="header-section-number">S.5</span> Discussion and Limitations</h1>
<p>The results presented in this paper provide evidence
of an immune phenotype in SKCM that corresponds to poor prognosis from a number of different perspectives. In terms of survival, we have demonstrated that adjusting for other relevant clinical factors, an increase in average logit-IP is associated with an increased hazard of death. In terms of gene expression data, average logit-IP is associated with decreased expression of genes related to immune cells that are generally associated with good overall prognosis. And finally, we found that average logit-IP is significantly lower in biopsies that have been classified as immune enriched through gene expression clustering, as well as biopsies that have been assessed by pathologists to have higher densities and spatial distributions of tumor infiltrating lymphocytes. These results fundamentally support the notion that CTIP is capturing an anti-immunogenic phenotype that is associated with poor prognosis, possibly due to an overall negative association with the presence of immune cells that are associated with improved prognosis such as tumor infiltrating lymphocytes and natural killer cells.</p>
<p>A fundamental limitation of this analysis is that immune cells were modeled as a single class rather than as separate cell types belonging to the same family. While the results paint a consistent picture from several different angles, the exact mechanism of these various associations remains fundamentally opaque because of this. It is worth emphasizing, however, that nothing about the structure of our pipeline depends on the presence of only one type of immune cell in addition to the cancer cells. Thus, while our present analysis raises questions about the impact of spatial immune activation on the progression of a tumor, it also offers the tools to further investigate such questions.</p>
<p>The SPARTIN pipeline represents a valuable contribution in and of
itself to digital pathology
through its ability to model and quantify immune infiltration
across entire biopsies in a way that captures meaningful variation
across patients. However, it also creates numerous opportunities
for future work unrelated to the present investigation. As algorithms for cell-level image classification improve, the
opportunities for more and more granular quantitative analysis of
histopathological imaging data will become both more numerous and
more fruitful. Specifically, as the ability to reliably distinguish between different types of immune cells will allow for more granular investigations into the prognoses associated with interaction between immune cells of different kinds and the tumor. Moreover, as spatial gene expression data becomes
more and more common, so too will opportunities for synthesizing
data on the relative spatial locations of different cell types
along with local gene expression data through complex modeling. It would be quite
trivial (particularly in a Bayesian framework) to model a function
of one or all of the parameters of interest as a linear combination
of the local gene expression data for genes that are known to be
relevant to immune response. Alternatively, future methods in this
area could be developed at the intersection of spatial data and
high dimensional data in order to identify such genes.</p>
<p>The SPARTIN pipeline also creates numerous opportunities for further investigation not directly related to assessing immune infiltration. Partitioning each biopsy into non-overlapping
sub-regions invites the application
of other tools from the spatial statistical canon to arbitrary spatial features of the tumor microenvironment. In fact, the primary feature of interest need not even be spatial in nature; so long as a feature (such as average tumor cell size) can be quantified at the cellular level and may be expected to vary across the tumor microenvironment, SPARTIN provides a framework for mapping the variation the variation of that feature across the entire biopsy.</p>
<p>There is another limitation of our analysis that suggests an avenue for future methodological development. Note that the result of fitting models on the partitioned tumor microenvironment is a set of repeated observations with some underlying spatial covariance structure related to the spatial relationships between the different subspaces. For now, we have ignored this spatial covariance, and instead chosen to summarize the variation at the biopsy level. While there is nothing intrinsically incorrect about this approach, it is quite likely that incorporating the spatial structure across the biopsy will lead to gains in efficiency, and possibly insight into the nature of the spatial variation in immune activation in the tumor microenvironment. Thus, future work will likely include the development of methods to efficiently analyze this complex structured areal data.</p>

<div id="refs" class="references csl-bib-body hanging-indent">
<div class="csl-entry">
Achkar, Tala, and Ahmad A Tarhini. 2017. <span>“The Use of Immunotherapy in the Treatment of Melanoma.”</span> <em>Journal of Hematology &amp; Oncology</em> 10 (1): 1–9.
</div>
<div class="csl-entry">
Akbani, Rehan, Kadir C Akdemir, B Arman Aksoy, Monique Albert, Adrian Ally, Samirkumar B Amin, Harindra Arachchi, et al. 2015. <span>“Genomic Classification of Cutaneous Melanoma.”</span> <em>Cell</em> 161 (7): 1681–96.
</div>
<div class="csl-entry">
Amgad, Mohamed, Anindya Sarkar, Chukka Srinivas, Rachel Redman, Simrath Ratra, Charles J Bechert, Benjamin C Calhoun, et al. 2019. <span>“Joint Region and Nucleus Segmentation for Characterization of Tumor Infiltrating Lymphocytes in Breast Cancer.”</span> In <em>Medical Imaging 2019: Digital Pathology</em>, 10956:129–36. SPIE.
</div>
<div class="csl-entry">
Baddeley, Adrian, and Rolf Turner. 2000. <span>“Practical Maximum Pseudolikelihood for Spatial Point Patterns: (With Discussion).”</span> <em>Australian &amp; New Zealand Journal of Statistics</em> 42 (3): 283–322.
</div>
<div class="csl-entry">
———. 2005. <span>“Spatstat: An r Package for Analyzing Spatial Point Patterns.”</span> <em>Journal of Statistical Software</em> 12: 1–42.
</div>
<div class="csl-entry">
Bareche, Yacine, Laurence Buisseret, Tina Gruosso, Edwina Girard, David Venet, Floriane Dupont, Christine Desmedt, et al. 2020. <span>“Unraveling Triple-Negative Breast Cancer Tumor Microenvironment Heterogeneity: Towards an Optimized Treatment Approach.”</span> <em>JNCI: Journal of the National Cancer Institute</em> 112 (7): 708–19.
</div>
<div class="csl-entry">
Becht, Etienne, Nicolas A. Giraldo, Laetitia Lacroix, Bénédicte Buttard, Nabila Elarouci, Florent Petitprez, Janick Selves, et al. 2016. <span>“Estimating the Population Abundance of Tissue-Infiltrating Immune and Stromal Cell Populations Using Gene Expression.”</span> <em>Genome Biology</em> 17 (1): 218. <a href="https://doi.org/10.1186/s13059-016-1070-5">https://doi.org/10.1186/s13059-016-1070-5</a>.
</div>
<div class="csl-entry">
Berger, Michael F, Eran Hodis, Timothy P Heffernan, Yonathan Lissanu Deribe, Michael S Lawrence, Alexei Protopopov, Elena Ivanova, et al. 2012. <span>“Melanoma Genome Sequencing Reveals Frequent Prex2 Mutations.”</span> <em>Nature</em> 485 (7399): 502–6.
</div>
<div class="csl-entry">
Bhattacharya, Sanchita, Patrick Dunn, Cristel G Thomas, Barry Smith, Henry Schaefer, Jieming Chen, Zicheng Hu, et al. 2018. <span>“ImmPort, Toward Repurposing of Open Access Immunological Assay Data for Translational and Clinical Research.”</span> <em>Scientific Data</em> 5 (1): 1–9.
</div>
<div class="csl-entry">
Bian, Yun, Yan Fang Liu, Hui Jiang, Yinghao Meng, Fang Liu, Kai Cao, Hao Zhang, et al. 2021. <span>“Machine Learning for MRI Radiomics: A Study Predicting Tumor-Infiltrating Lymphocytes in Patients with Pancreatic Ductal Adenocarcinoma.”</span> <em>Abdominal Radiology</em> 46 (10): 4800–4816.
</div>
<div class="csl-entry">
Boxberg, Melanie, Lena Leising, Katja Steiger, Moritz Jesinghaus, Aezlat Alkhamas, Marion Mielke, Nicole Pfarr, et al. 2019. <span>“Composition and Clinical Impact of the Immunologic Tumor Microenvironment in Oral Squamous Cell Carcinoma.”</span> <em>The Journal of Immunology</em> 202 (1): 278–91.
</div>
<div class="csl-entry">
Denkert, Carsten, Gunter Von Minckwitz, Jan C Brase, Bruno V Sinn, Stephan Gade, Ralf Kronenwett, Berit M Pfitzner, et al. 2015. <span>“Tumor-Infiltrating Lymphocytes and Response to Neoadjuvant Chemotherapy with or Without Carboplatin in Human Epidermal Growth Factor Receptor 2-Positive and Triple-Negative Primary Breast Cancers.”</span> <em>J Clin Oncol</em> 33 (9): 983–91.
</div>
<div class="csl-entry">
Dieci, Maria Vittoria, Nina Radosevic-Robin, Susan Fineberg, Gert Van den Eynden, Nils Ternes, Frédérique Penault-Llorca, Giancarlo Pruneri, et al. 2018. <span>“Update on Tumor-Infiltrating Lymphocytes (TILs) in Breast Cancer, Including Recommendations to Assess TILs in Residual Disease After Neoadjuvant Therapy and in Carcinoma in Situ: A Report of the International Immuno-Oncology Biomarker Working Group on Breast Cancer.”</span> In <em>Seminars in Cancer Biology</em>, 52:16–25. Elsevier.
</div>
<div class="csl-entry">
Franklin, C, E Livingstone, A Roesch, B Schilling, and D Schadendorf. 2017. <span>“Immunotherapy in Melanoma: Recent Advances and Future Directions.”</span> <em>European Journal of Surgical Oncology (EJSO)</em> 43 (3): 604–11.
</div>
<div class="csl-entry">
Fu, Qiaofen, Nan Chen, Chunlei Ge, Ruilei Li, Zhen Li, Baozhen Zeng, Chunyan Li, et al. 2019. <span>“Prognostic Value of Tumor-Infiltrating Lymphocytes in Melanoma: A Systematic Review and Meta-Analysis.”</span> <em>Oncoimmunology</em> 8 (7): e1593806.
</div>
<div class="csl-entry">
Heindl, Andreas, Sidra Nawaz, and Yinyin Yuan. 2015. <span>“Mapping Spatial Heterogeneity in the Tumor Microenvironment: A New Era for Digital Pathology.”</span> <em>Laboratory Investigation</em> 95 (4): 377–84.
</div>
<div class="csl-entry">
Högmander, Harri, and Aila Särkkä. 1999. <span>“Multitype Spatial Point Patterns with Hierarchical Interactions.”</span> <em>Biometrics</em> 55 (4): 1051–58.
</div>
<div class="csl-entry">
Hunter, Miranda V, Reuben Moncada, Joshua M Weiss, Itai Yanai, and Richard M White. 2021. <span>“Spatially Resolved Transcriptomics Reveals the Architecture of the Tumor-Microenvironment Interface.”</span> <em>Nature Communications</em> 12 (1): 1–16.
</div>
<div class="csl-entry">
Idos, Gregory E, Janet Kwok, Nirupama Bonthala, Lynn Kysh, Stephen B Gruber, and Chenxu Qu. 2020. <span>“The Prognostic Implications of Tumor Infiltrating Lymphocytes in Colorectal Cancer: A Systematic Review and Meta-Analysis.”</span> <em>Scientific Reports</em> 10 (1): 1–14.
</div>
<div class="csl-entry">
Kang, Jian, Timothy D Johnson, Thomas E Nichols, and Tor D Wager. 2011. <span>“Meta Analysis of Functional Neuroimaging Data via Bayesian Spatial Point Processes.”</span> <em>Journal of the American Statistical Association</em> 106 (493): 124–34.
</div>
<div class="csl-entry">
King, Ruth, Janine B Illian, Stuart E King, Glenna F Nightingale, and Ditte K Hendrichsen. 2012. <span>“A Bayesian Approach to Fitting Gibbs Processes with Temporal Random Effects.”</span> <em>Journal of Agricultural, Biological, and Environmental Statistics</em> 17 (4): 601–22.
</div>
<div class="csl-entry">
Komura, Daisuke, and Shumpei Ishikawa. 2018. <span>“Machine Learning Methods for Histopathological Image Analysis.”</span> <em>Computational and Structural Biotechnology Journal</em> 16: 34–42.
</div>
<div class="csl-entry">
Li, Qiwei, Xinlei Wang, Faming Liang, and Guanghua Xiao. 2019. <span>“A Bayesian Mark Interaction Model for Analysis of Tumor Pathology Images.”</span> <em>The Annals of Applied Statistics</em> 13 (3): 1708.
</div>
<div class="csl-entry">
Li, Taiwen, Jingxin Fu, Zexian Zeng, David Cohen, Jing Li, Qianming Chen, Bo Li, and X Shirley Liu. 2020. <span>“Timer2.0 for Analysis of Tumor-Infiltrating Immune Cells.”</span> <em>Nucleic Acids Research</em> 48 (W1): W509–14. <a href="https://doi.org/10.1093/nar/gkaa407">https://doi.org/10.1093/nar/gkaa407</a>.
</div>
<div class="csl-entry">
Lu, Zixiao, Siwen Xu, Wei Shao, Yi Wu, Jie Zhang, Zhi Han, Qianjin Feng, and Kun Huang. 2020. <span>“Deep-Learning–Based Characterization of Tumor-Infiltrating Lymphocytes in Breast Cancers from Histopathology Images and Multiomics Data.”</span> <em>JCO Clinical Cancer Informatics</em> 4: 480–90.
</div>
<div class="csl-entry">
Negahbani, Farzin, Rasool Sabzi, Bita Pakniyat Jahromi, Dena Firouzabadi, Fateme Movahedi, Mahsa Kohandel Shirazi, Shayan Majidi, and Amirreza Dehghanian. 2021. <span>“PathoNet Introduced as a Deep Neural Network Backend for Evaluation of Ki-67 and Tumor-Infiltrating Lymphocytes in Breast Cancer.”</span> <em>Scientific Reports</em> 11 (1): 1–13.
</div>
<div class="csl-entry">
Pagès, F, J Galon, M-C Dieu-Nosjean, E Tartour, C Sautès-Fridman, and W-H Fridman. 2010. <span>“Immune Infiltration in Human Tumors: A Prognostic Factor That Should Not Be Ignored.”</span> <em>Oncogene</em> 29 (8): 1093–1102. <a href="https://doi.org/10.1038/onc.2009.416">https://doi.org/10.1038/onc.2009.416</a>.
</div>
<div class="csl-entry">
R Core Team. 2021. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div class="csl-entry">
Ray, Meredith, Jian Kang, and Hongmei Zhang. 2015. <span>“Identifying Activation Centers with Spatial Cox Point Processes Using Fmri Data.”</span> <em>IEEE/ACM Transactions on Computational Biology and Bioinformatics</em> 13 (6): 1130–41.
</div>
<div class="csl-entry">
Saltz, Joel, Rajarsi Gupta, Le Hou, Tahsin Kurc, Pankaj Singh, Vu Nguyen, Dimitris Samaras, et al. 2018. <span>“Spatial Organization and Molecular Correlation of Tumor-Infiltrating Lymphocytes Using Deep Learning on Pathology Images.”</span> <em>Cell Reports</em> 23 (1): 181–93.
</div>
<div class="csl-entry">
Sica, Antonio, Paola Allavena, and Alberto Mantovani. 2008. <span>“Cancer Related Inflammation: The Macrophage Connection.”</span> <em>Cancer Letters</em> 267 (2): 204–15. <a href="https://doi.org/10.1016/j.canlet.2008.03.028">https://doi.org/10.1016/j.canlet.2008.03.028</a>.
</div>
<div class="csl-entry">
Strauss, David J. 1975. <span>“A Model for Clustering.”</span> <em>Biometrika</em> 62 (2): 467–75.
</div>
<div class="csl-entry">
Sturm, Gregor, Francesca Finotello, Florent Petitprez, Jitao David Zhang, Jan Baumbach, Wolf H Fridman, Markus List, and Tatsiana Aneichyk. 2019. <span>“Comprehensive Evaluation of Transcriptome-Based Cell-Type Quantification Methods for Immuno-Oncology.”</span> <em>Bioinformatics</em> 35 (14): i436–45. <a href="https://doi.org/10.1093/bioinformatics/btz363">https://doi.org/10.1093/bioinformatics/btz363</a>.
</div>
<div class="csl-entry">
Su, Yu-Sung, Masanao Yajima, Maintainer Yu-Sung Su, and JAGS SystemRequirements. 2015. <span>“Package <span>‘R2jags’</span>.”</span> <em>R Package Version 0.03-08, URL Http://CRAN. R-Project. Org/Package= R2jags</em>.
</div>
<div class="csl-entry">
Tsutsui, Shinichi, Kazuhiro Yasuda, Kosuke Suzuki, Kouichirou Tahara, Hidefumi Higashi, and Shoichi Era. 2005. <span>“Macrophage Infiltration and Its Prognostic Implications in Breast Cancer: The Relationship with VEGF Expression and Microvessel Density.”</span> <em>Oncology Reports</em>, August. <a href="https://doi.org/10.3892/or.14.2.425">https://doi.org/10.3892/or.14.2.425</a>.
</div>
<div class="csl-entry">
Wang, Shidan, Tao Wang, Lin Yang, Donghan M Yang, Junya Fujimoto, Faliu Yi, Xin Luo, et al. 2019. <span>“ConvPath: A Software Tool for Lung Adenocarcinoma Digital Pathological Image Analysis Aided by a Convolutional Neural Network.”</span> <em>EBioMedicine</em> 50: 103–10.
</div>
<div class="csl-entry">
Yan, Yiyi, Alexey A Leontovich, Michael J Gerdes, Keyur Desai, Jinhong Dong, Anup Sood, Alberto Santamaria-Pang, et al. 2019. <span>“Understanding Heterogeneous Tumor Microenvironment in Metastatic Melanoma.”</span> <em>PloS One</em> 14 (6): e0216485.
</div>
<div class="csl-entry">
Yuan, Yinyin. 2016. <span>“Spatial Heterogeneity in the Tumor Microenvironment.”</span> <em>Cold Spring Harbor Perspectives in Medicine</em> 6 (8): a026583.
</div>
<div class="csl-entry">
Zhu, Yitan, Peng Qiu, and Yuan Ji. 2014. <span>“TCGA-Assembler: Open-Source Software for Retrieving and Processing TCGA Data.”</span> <em>Nature Methods</em> 11 (6): 599–600.
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="results.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
