<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>S.5 Genomic Associations | SPARTIN Supplementary material</title>
  <meta name="description" content="Supplementary material related to the SPARTIN pipeline" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="S.5 Genomic Associations | SPARTIN Supplementary material" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Supplementary material related to the SPARTIN pipeline" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="S.5 Genomic Associations | SPARTIN Supplementary material" />
  
  <meta name="twitter:description" content="Supplementary material related to the SPARTIN pipeline" />
  

<meta name="author" content="Nate Osher" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cell-classification-model.html"/>
<link rel="next" href="association-with-deconvolution-data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SPARTIN Supplement</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="model-fitting.html"><a href="model-fitting.html"><i class="fa fa-check"></i><b>2</b> Model Fitting</a></li>
<li class="chapter" data-level="3" data-path="biopsy-partition.html"><a href="biopsy-partition.html"><i class="fa fa-check"></i><b>3</b> Biopsy Partition</a></li>
<li class="chapter" data-level="4" data-path="cell-classification-model.html"><a href="cell-classification-model.html"><i class="fa fa-check"></i><b>4</b> Cell Classification Model</a>
<ul>
<li class="chapter" data-level="4.1" data-path="cell-classification-model.html"><a href="cell-classification-model.html#whole-slide-image-wsi-retrieval-and-obtaining-training-labels"><i class="fa fa-check"></i><b>4.1</b> Whole-Slide Image (WSI) Retrieval and Obtaining Training Labels</a></li>
<li class="chapter" data-level="4.2" data-path="cell-classification-model.html"><a href="cell-classification-model.html#whole-slide-image-wsi-nucleus-segmentation"><i class="fa fa-check"></i><b>4.2</b> Whole-Slide Image (WSI) Nucleus Segmentation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genomic-associations.html"><a href="genomic-associations.html"><i class="fa fa-check"></i><b>5</b> Genomic Associations</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genomic-associations.html"><a href="genomic-associations.html#association-with-expression-of-significantly-mutated-genes"><i class="fa fa-check"></i><b>5.1</b> Association with expression of significantly mutated genes</a></li>
<li class="chapter" data-level="5.2" data-path="genomic-associations.html"><a href="genomic-associations.html#association-with-expression-of-immune-genes"><i class="fa fa-check"></i><b>5.2</b> Association with expression of immune genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="association-with-deconvolution-data.html"><a href="association-with-deconvolution-data.html"><i class="fa fa-check"></i><b>6</b> Association with Deconvolution Data</a></li>
<li class="chapter" data-level="7" data-path="spartin-r-package.html"><a href="spartin-r-package.html"><i class="fa fa-check"></i><b>7</b> SPARTIN R package</a>
<ul>
<li class="chapter" data-level="7.1" data-path="spartin-r-package.html"><a href="spartin-r-package.html#simulation-and-model-fitting"><i class="fa fa-check"></i><b>7.1</b> Simulation and Model Fitting</a></li>
<li class="chapter" data-level="7.2" data-path="spartin-r-package.html"><a href="spartin-r-package.html#estimating-ctip"><i class="fa fa-check"></i><b>7.2</b> Estimating CTIP</a></li>
<li class="chapter" data-level="7.3" data-path="spartin-r-package.html"><a href="spartin-r-package.html#tesselation-and-intensity-thresholding"><i class="fa fa-check"></i><b>7.3</b> Tesselation and Intensity Thresholding</a></li>
<li class="chapter" data-level="7.4" data-path="spartin-r-package.html"><a href="spartin-r-package.html#interactive-visualization"><i class="fa fa-check"></i><b>7.4</b> Interactive Visualization</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">SPARTIN Supplementary material</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genomic-associations" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">S.5</span> Genomic Associations<a href="genomic-associations.html#genomic-associations" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="association-with-expression-of-significantly-mutated-genes" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Association with expression of significantly mutated genes<a href="genomic-associations.html#association-with-expression-of-significantly-mutated-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In addition to associations with survival, we investigated the
association between our measurement and gene expression. Gene
expression data was acquired for all 335 patients in
<!-- A quick note about citations here: the Zhu one is the paper -->
<!-- I meant to cite- that paper lays out the TCGA assembler and -->
<!-- talks about the specific modules I used to get the data. -->
<!-- Re: Akbani, that's a little more complicated. See: -->
<!-- https://scholar.google.com/scholar?hl=en&q=genomic+classification+of+cutaneous+melanoma -->
<!-- This yields two different citations for (as far as I can tell -->
<!-- the same paper; the more utilized one is the one I included.  -->
our sample using TCGA Assembler (<span class="citation">Zhu, Qiu, and Ji (<a href="#ref-Zhu14" role="doc-biblioref">2014</a>)</span>). Additionally,
42 significantly mutated genes (SMGs) of interest were identified
using previous work investigating the genomic differences in
SKCM (<span class="citation">Akbani et al. (<a href="#ref-Akbani15" role="doc-biblioref">2015</a>)</span>).</p>
<p>Of the 335 patients in our sample, 330 had gene expression data
for all genes of interest, while 95 were missing data for all
genes of interest. We examined the marginal association between
the normalized gene expression data for the 330 patients with
complete data and average logit CTIP values. Marginal association
was assessed via univariate simple linear regression, carried out
separately for each gene. The Wald statistic of the coefficient
corresponding to gene expression was used to produce a <span class="math inline">\(p\)</span>-value.
After correcting for multiple testing using a Bonferroni correction with <span class="math inline">\(\alpha = 0.05\)</span>, we found that LRRC37A3 was significantly positively associated with
average logit CTIP (<span class="math inline">\(p &lt;&lt; 0.0001\)</span>). See Table 5.2 below for the full list of genes, significance of associations, and directionality of associations.</p>
<table>
<thead>
<tr>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:left;">
P-value
</th>
<th style="text-align:left;">
Direction of Association
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
LRRC37A3
</td>
<td style="text-align:left;">
3.9e-05 *
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
B2M
</td>
<td style="text-align:left;">
0.0012
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
TP53
</td>
<td style="text-align:left;">
0.0019
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
BRAF
</td>
<td style="text-align:left;">
0.024
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
RCAN2
</td>
<td style="text-align:left;">
0.027
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
TMEM216
</td>
<td style="text-align:left;">
0.032
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
ACD
</td>
<td style="text-align:left;">
0.033
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
PPIAL4G
</td>
<td style="text-align:left;">
0.043
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
TBC1D3B
</td>
<td style="text-align:left;">
0.052
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
FAM113B
</td>
<td style="text-align:left;">
0.055
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
COL9A2
</td>
<td style="text-align:left;">
0.057
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
SIRPB1
</td>
<td style="text-align:left;">
0.064
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
ZFX
</td>
<td style="text-align:left;">
0.073
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
RB1
</td>
<td style="text-align:left;">
0.087
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
DDX3X
</td>
<td style="text-align:left;">
0.1
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
RQCD1
</td>
<td style="text-align:left;">
0.13
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
CCDC28A
</td>
<td style="text-align:left;">
0.14
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
NDUFB9
</td>
<td style="text-align:left;">
0.15
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
EMG1
</td>
<td style="text-align:left;">
0.15
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
ITGA4
</td>
<td style="text-align:left;">
0.18
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
OXA1L
</td>
<td style="text-align:left;">
0.19
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
CTNNB1
</td>
<td style="text-align:left;">
0.2
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
STK19
</td>
<td style="text-align:left;">
0.27
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
MAP2K1
</td>
<td style="text-align:left;">
0.28
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
NRAS
</td>
<td style="text-align:left;">
0.28
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
GNAI2
</td>
<td style="text-align:left;">
0.32
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
RAPGEF5
</td>
<td style="text-align:left;">
0.36
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
NF1
</td>
<td style="text-align:left;">
0.37
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
IDH1
</td>
<td style="text-align:left;">
0.37
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
PTEN
</td>
<td style="text-align:left;">
0.4
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
ARID2
</td>
<td style="text-align:left;">
0.42
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
RAC1
</td>
<td style="text-align:left;">
0.42
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
CDKN2A
</td>
<td style="text-align:left;">
0.56
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
MRPS31
</td>
<td style="text-align:left;">
0.64
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
MSR1
</td>
<td style="text-align:left;">
0.7
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
WDR12
</td>
<td style="text-align:left;">
0.72
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
NOTCH2NL
</td>
<td style="text-align:left;">
0.83
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
PPP6C
</td>
<td style="text-align:left;">
0.85
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
C3orf71
</td>
<td style="text-align:left;">
0.89
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
RPS27
</td>
<td style="text-align:left;">
0.9
</td>
<td style="text-align:left;">
Negative
</td>
</tr>
<tr>
<td style="text-align:left;">
FAM58A
</td>
<td style="text-align:left;">
0.95
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
<tr>
<td style="text-align:left;">
KNSTRN
</td>
<td style="text-align:left;">
0.98
</td>
<td style="text-align:left;">
Positive
</td>
</tr>
</tbody>
<tfoot>
<tr>
<td style="padding: 0; border:0;" colspan="100%">
<sup></sup> Table 5.2: Significantly mutated genes, unadjusted p-values, and direction of association with biopsy level average logit CTIP. Asterisk indicates significance under Bonferroni correction, alpha = 0.05.
</td>
</tr>
</tfoot>
</table>
</div>
<div id="association-with-expression-of-immune-genes" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Association with expression of immune genes<a href="genomic-associations.html#association-with-expression-of-immune-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We also assessed the association between CTIP and genes that are
associated with immune activity. <span class="citation">Bhattacharya et al. (<a href="#ref-Bhattacharya18" role="doc-biblioref">2018</a>)</span> have collected and classified a list of 1,793 unique genes associated with various aspects of human immune activity. Of these, gene expression data was available for 1,305 genes across the same 330 patients used in the previous gene expression analysis. We assessed the univariate association between biopsy level mean logit CTIP and these genes using Spearman Correlation. The advantage of Spearman correlation as opposed to the more standard Pearson correlation is that the former does not assume a linear relationship between the underlying variables of interest. Such assumptions can be problematic, particularly when there is no strong reason a priori to believe the relationship between the two variables is of a particular form. However, like Pearson correlation Spearman correlation is defined to lie in <span class="math inline">\([-1,1]\)</span>, with each extreme indicating the same directionality and strength of association as Pearson Correlation. Finally, statistical significance of associations was calculated using the cor.test function of the R programming language (<span class="citation">R Core Team (<a href="#ref-Rteam" role="doc-biblioref">2021</a>)</span>). After applying a Bonferroni correction, we found that 28 genes were significantly associated with IP. For the complete list of genes, see Table 5.2 below.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:04"></span>
<img src="Figures/IMMPORT%20Spearman.png" alt="$-log_{10}$ p-values plotted against Spearman Correlation. Colors indicate Group as determined by IMMPORT; Red line indicates Bonferroni correction" width="100%" />
<p class="caption">
Figure 5.1: <span class="math inline">\(-log_{10}\)</span> p-values plotted against Spearman Correlation. Colors indicate Group as determined by IMMPORT; Red line indicates Bonferroni correction
</p>
</div>
<table>
<thead>
<tr>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:left;">
Group
</th>
<th style="text-align:left;">
Spearman Correlation
</th>
<th style="text-align:left;">
P-Value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
CD244
</td>
<td style="text-align:left;">
Natural Killer Cell
</td>
<td style="text-align:left;">
-0.29
</td>
<td style="text-align:left;">
&lt;&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
CTSS
</td>
<td style="text-align:left;">
Antimicrobials
</td>
<td style="text-align:left;">
-0.29
</td>
<td style="text-align:left;">
&lt;&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
CCR8
</td>
<td style="text-align:left;">
Cytokine Receptors
</td>
<td style="text-align:left;">
-0.27
</td>
<td style="text-align:left;">
&lt;&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
CD3G
</td>
<td style="text-align:left;">
TCR Signaling Pathway
</td>
<td style="text-align:left;">
-0.26
</td>
<td style="text-align:left;">
&lt;&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
PTPRC
</td>
<td style="text-align:left;">
TCR Signaling Pathway
</td>
<td style="text-align:left;">
-0.26
</td>
<td style="text-align:left;">
&lt;&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
TLR8
</td>
<td style="text-align:left;">
Antimicrobials
</td>
<td style="text-align:left;">
-0.26
</td>
<td style="text-align:left;">
&lt;&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
ITK
</td>
<td style="text-align:left;">
TCR Signaling Pathway
</td>
<td style="text-align:left;">
-0.26
</td>
<td style="text-align:left;">
&lt;&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
TLR3
</td>
<td style="text-align:left;">
Antimicrobials
</td>
<td style="text-align:left;">
-0.25
</td>
<td style="text-align:left;">
&lt;&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
STAT1
</td>
<td style="text-align:left;">
Antimicrobials
</td>
<td style="text-align:left;">
-0.25
</td>
<td style="text-align:left;">
&lt;&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
TNFRSF9
</td>
<td style="text-align:left;">
TNF Family Members Receptors
</td>
<td style="text-align:left;">
-0.25
</td>
<td style="text-align:left;">
&lt;&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
EREG
</td>
<td style="text-align:left;">
Cytokines
</td>
<td style="text-align:left;">
0.24
</td>
<td style="text-align:left;">
&lt;&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
LCK
</td>
<td style="text-align:left;">
TCR Signaling Pathway
</td>
<td style="text-align:left;">
-0.24
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
PIK3CG
</td>
<td style="text-align:left;">
TCR Signaling Pathway
</td>
<td style="text-align:left;">
-0.24
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
ERAP1
</td>
<td style="text-align:left;">
Antigen Processing and Presentation
</td>
<td style="text-align:left;">
-0.24
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
TNFSF14
</td>
<td style="text-align:left;">
TNF Family Members
</td>
<td style="text-align:left;">
-0.23
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
IL6R
</td>
<td style="text-align:left;">
Interleukins Receptors
</td>
<td style="text-align:left;">
-0.23
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
CIITA
</td>
<td style="text-align:left;">
Antigen Processing and Presentation
</td>
<td style="text-align:left;">
-0.23
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
ICOS
</td>
<td style="text-align:left;">
TCR Signaling Pathway
</td>
<td style="text-align:left;">
-0.23
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
DES
</td>
<td style="text-align:left;">
Antimicrobials
</td>
<td style="text-align:left;">
0.23
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
APLN
</td>
<td style="text-align:left;">
Cytokines
</td>
<td style="text-align:left;">
0.23
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
IFIH1
</td>
<td style="text-align:left;">
Antimicrobials
</td>
<td style="text-align:left;">
-0.23
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
RFX5
</td>
<td style="text-align:left;">
Antigen Processing and Presentation
</td>
<td style="text-align:left;">
-0.23
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
LYZ
</td>
<td style="text-align:left;">
Antimicrobials
</td>
<td style="text-align:left;">
-0.23
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
CXCR6
</td>
<td style="text-align:left;">
Cytokine Receptors
</td>
<td style="text-align:left;">
-0.23
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
CRABP1
</td>
<td style="text-align:left;">
Antimicrobials
</td>
<td style="text-align:left;">
0.23
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
CYBB
</td>
<td style="text-align:left;">
Antimicrobials
</td>
<td style="text-align:left;">
-0.23
</td>
<td style="text-align:left;">
&lt; 0.0001
</td>
</tr>
</tbody>
<tfoot>
<tr>
<td style="padding: 0; border:0;" colspan="100%">
<sup></sup> Table 5.2: IMMPORT Genes significantly associated with average logit biopsy level CTIP after Bonferroni correction, alpha = 0.05.
</td>
</tr>
</tfoot>
</table>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Akbani15" class="csl-entry">
Akbani, Rehan, Kadir C Akdemir, B Arman Aksoy, Monique Albert, Adrian Ally, Samirkumar B Amin, Harindra Arachchi, et al. 2015. <span>“Genomic Classification of Cutaneous Melanoma.”</span> <em>Cell</em> 161 (7): 1681–96.
</div>
<div id="ref-Bhattacharya18" class="csl-entry">
Bhattacharya, Sanchita, Patrick Dunn, Cristel G Thomas, Barry Smith, Henry Schaefer, Jieming Chen, Zicheng Hu, et al. 2018. <span>“ImmPort, Toward Repurposing of Open Access Immunological Assay Data for Translational and Clinical Research.”</span> <em>Scientific Data</em> 5 (1): 1–9.
</div>
<div id="ref-Rteam" class="csl-entry">
R Core Team. 2021. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-Zhu14" class="csl-entry">
Zhu, Yitan, Peng Qiu, and Yuan Ji. 2014. <span>“TCGA-Assembler: Open-Source Software for Retrieving and Processing TCGA Data.”</span> <em>Nature Methods</em> 11 (6): 599–600.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cell-classification-model.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="association-with-deconvolution-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
