<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>S.9 Introduction | SPARTIN Supplementary material</title>
  <meta name="description" content="Supplementary material related to the SPARTYN pipeline" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="S.9 Introduction | SPARTIN Supplementary material" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Supplementary material related to the SPARTYN pipeline" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="S.9 Introduction | SPARTIN Supplementary material" />
  
  <meta name="twitter:description" content="Supplementary material related to the SPARTYN pipeline" />
  

<meta name="author" content="Nate Osher" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="preface-1.html"/>
<link rel="next" href="materials-and-methods.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="model-fitting.html"><a href="model-fitting.html"><i class="fa fa-check"></i><b>2</b> Model Fitting</a></li>
<li class="chapter" data-level="3" data-path="biopsy-partition.html"><a href="biopsy-partition.html"><i class="fa fa-check"></i><b>3</b> Biopsy Partition</a></li>
<li class="chapter" data-level="4" data-path="cell-classification-model.html"><a href="cell-classification-model.html"><i class="fa fa-check"></i><b>4</b> Cell Classification Model</a>
<ul>
<li class="chapter" data-level="4.1" data-path="cell-classification-model.html"><a href="cell-classification-model.html#whole-slide-image-wsi-retrieval-and-obtaining-training-labels"><i class="fa fa-check"></i><b>4.1</b> Whole-Slide Image (WSI) Retrieval and Obtaining Training Labels</a></li>
<li class="chapter" data-level="4.2" data-path="cell-classification-model.html"><a href="cell-classification-model.html#whole-slide-image-wsi-nucleus-segmentation"><i class="fa fa-check"></i><b>4.2</b> Whole-Slide Image (WSI) Nucleus Segmentation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genomic-associations.html"><a href="genomic-associations.html"><i class="fa fa-check"></i><b>5</b> Genomic Associations</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genomic-associations.html"><a href="genomic-associations.html#association-with-expression-of-significantly-mutated-genes"><i class="fa fa-check"></i><b>5.1</b> Association with expression of significantly mutated genes</a></li>
<li class="chapter" data-level="5.2" data-path="genomic-associations.html"><a href="genomic-associations.html#association-with-expression-of-immune-genes"><i class="fa fa-check"></i><b>5.2</b> Association with expression of immune genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="association-with-deconvolution-data.html"><a href="association-with-deconvolution-data.html"><i class="fa fa-check"></i><b>6</b> Association with Deconvolution Data</a></li>
<li class="chapter" data-level="7" data-path="spartin-r-package.html"><a href="spartin-r-package.html"><i class="fa fa-check"></i><b>7</b> SPARTIN R package</a></li>
<li class="chapter" data-level="8" data-path="preface-1.html"><a href="preface-1.html"><i class="fa fa-check"></i><b>8</b> Preface</a></li>
<li class="chapter" data-level="9" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>9</b> Introduction</a></li>
<li class="chapter" data-level="10" data-path="materials-and-methods.html"><a href="materials-and-methods.html"><i class="fa fa-check"></i><b>10</b> Materials and Methods</a>
<ul>
<li class="chapter" data-level="10.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#overview"><i class="fa fa-check"></i><b>10.1</b> Overview</a></li>
<li class="chapter" data-level="10.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#spatial-point-process-models"><i class="fa fa-check"></i><b>10.2</b> Spatial Point Process Models</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#data-structure"><i class="fa fa-check"></i><b>10.2.1</b> Data Structure</a></li>
<li class="chapter" data-level="10.2.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#model"><i class="fa fa-check"></i><b>10.2.2</b> Model</a></li>
<li class="chapter" data-level="10.2.3" data-path="materials-and-methods.html"><a href="materials-and-methods.html#interpretation"><i class="fa fa-check"></i><b>10.2.3</b> Interpretation</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="materials-and-methods.html"><a href="materials-and-methods.html#pseudolikelihood-function"><i class="fa fa-check"></i><b>10.3</b> Pseudolikelihood Function</a></li>
<li class="chapter" data-level="10.4" data-path="materials-and-methods.html"><a href="materials-and-methods.html#intensity-thresholding-and-partition"><i class="fa fa-check"></i><b>10.4</b> Intensity Thresholding and Partition</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#motivation-for-partition"><i class="fa fa-check"></i><b>10.4.1</b> Motivation for partition</a></li>
<li class="chapter" data-level="10.4.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#partition-pipeline"><i class="fa fa-check"></i><b>10.4.2</b> Partition pipeline</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="materials-and-methods.html"><a href="materials-and-methods.html#cell-type-interaction-probability"><i class="fa fa-check"></i><b>10.5</b> Cell Type Interaction Probability</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#motivation"><i class="fa fa-check"></i><b>10.5.1</b> Motivation</a></li>
<li class="chapter" data-level="10.5.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#definition"><i class="fa fa-check"></i><b>10.5.2</b> Definition</a></li>
<li class="chapter" data-level="10.5.3" data-path="materials-and-methods.html"><a href="materials-and-methods.html#estimating-the-empirical-null-distribution"><i class="fa fa-check"></i><b>10.5.3</b> Estimating the empirical null distribution</a></li>
<li class="chapter" data-level="10.5.4" data-path="materials-and-methods.html"><a href="materials-and-methods.html#computation-of-ctip"><i class="fa fa-check"></i><b>10.5.4</b> Computation of CTIP</a></li>
<li class="chapter" data-level="10.5.5" data-path="materials-and-methods.html"><a href="materials-and-methods.html#interpretation-1"><i class="fa fa-check"></i><b>10.5.5</b> Interpretation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="simulation.html"><a href="simulation.html"><i class="fa fa-check"></i><b>11</b> Simulation</a>
<ul>
<li class="chapter" data-level="11.1" data-path="simulation.html"><a href="simulation.html#simulation-overview"><i class="fa fa-check"></i><b>11.1</b> Simulation Overview</a></li>
<li class="chapter" data-level="11.2" data-path="simulation.html"><a href="simulation.html#negative-simulations"><i class="fa fa-check"></i><b>11.2</b> Negative simulations</a></li>
<li class="chapter" data-level="11.3" data-path="simulation.html"><a href="simulation.html#positive-simulations"><i class="fa fa-check"></i><b>11.3</b> Positive simulations</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="application.html"><a href="application.html"><i class="fa fa-check"></i><b>12</b> Application</a>
<ul>
<li class="chapter" data-level="12.1" data-path="application.html"><a href="application.html#background"><i class="fa fa-check"></i><b>12.1</b> Background</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="application.html"><a href="application.html#data-and-pre-processing"><i class="fa fa-check"></i><b>12.1.1</b> Data and Pre-Processing</a></li>
<li class="chapter" data-level="12.1.2" data-path="application.html"><a href="application.html#cell-classification-model-1"><i class="fa fa-check"></i><b>12.1.2</b> Cell Classification Model</a></li>
<li class="chapter" data-level="12.1.3" data-path="application.html"><a href="application.html#melanoma"><i class="fa fa-check"></i><b>12.1.3</b> Melanoma</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="application.html"><a href="application.html#genomic-associations-1"><i class="fa fa-check"></i><b>12.2</b> Genomic Associations</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="application.html"><a href="application.html#association-with-expression-of-significantly-mutated-genes-1"><i class="fa fa-check"></i><b>12.2.1</b> Association with expression of significantly mutated genes</a></li>
<li class="chapter" data-level="12.2.2" data-path="application.html"><a href="application.html#association-with-expression-of-immune-genes-1"><i class="fa fa-check"></i><b>12.2.2</b> Association with expression of immune genes</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="application.html"><a href="application.html#association-with-deconvolution-data-1"><i class="fa fa-check"></i><b>12.3</b> Association with Deconvolution Data</a></li>
<li class="chapter" data-level="12.4" data-path="application.html"><a href="application.html#association-with-other-outcomes-of-interest"><i class="fa fa-check"></i><b>12.4</b> Association with Other Outcomes of Interest</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="application.html"><a href="application.html#transcriptomic-classes"><i class="fa fa-check"></i><b>12.4.1</b> Transcriptomic classes</a></li>
<li class="chapter" data-level="12.4.2" data-path="application.html"><a href="application.html#pathologist-assessment"><i class="fa fa-check"></i><b>12.4.2</b> Pathologist assessment</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="application.html"><a href="application.html#survival-analysis"><i class="fa fa-check"></i><b>12.5</b> Survival Analysis</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="discussion-and-limitations.html"><a href="discussion-and-limitations.html"><i class="fa fa-check"></i><b>13</b> Discussion and Limitations</a>
<ul>
<li class="chapter" data-level="13.1" data-path="discussion-and-limitations.html"><a href="discussion-and-limitations.html#r-markdown"><i class="fa fa-check"></i><b>13.1</b> R Markdown</a></li>
<li class="chapter" data-level="13.2" data-path="discussion-and-limitations.html"><a href="discussion-and-limitations.html#including-plots"><i class="fa fa-check"></i><b>13.2</b> Including Plots</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">SPARTIN Supplementary material</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introduction" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">S.9</span> Introduction<a href="introduction.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>While the staining and examination of tissue samples has been a
ubiquitous medical practice for decades, it
has only been relatively recently that high definition images of
such stainings have begun to be explored through the lens of
machine learning and statistical modeling. A considerable amount
of work has thus far been put into building and training models
capable of accurate classifications of overall tissue
samples as well as subsections of tissue samples (<span class="citation">Saltz et al. (<a href="#ref-Saltz18" role="doc-biblioref">2018</a>)</span>; <span class="citation">Amgad et al. (<a href="#ref-Amgad19" role="doc-biblioref">2019</a>)</span>; <span class="citation">Lu et al. (<a href="#ref-Lu20" role="doc-biblioref">2020</a>)</span>;
<span class="citation">Negahbani et al. (<a href="#ref-Negahbani21" role="doc-biblioref">2021</a>)</span>; <span class="citation">Bian et al. (<a href="#ref-Bian21" role="doc-biblioref">2021</a>)</span>). This work, often broadly
referred to as Histopathological Image Analysis or Digital
Pathology, encompasses a wide range of methods and goals ranging
from developing models to assist in the scoring and staging of
cancer biopsies to the classification of cells in tumor biopsies (<span class="citation">Komura and Ishikawa (<a href="#ref-Komura18" role="doc-biblioref">2018</a>)</span>). This work has important implications for
biomedical research. Because of the precision with
which such algorithms can assess biopsies at the cellular level,
it is possible to uncover structures and associations not immediately apparent to human pathologists. This degree of precision allows for
the rigorous modeling and examination of intratumoral heterogeneity
with respect to arbitrary factors and patient characteristics.
These methods also allow for the assessment of specific features
of pathology images in a volume previously only available via the
time and energy of a substantial number of trained pathologists.</p>
<p>While valuable in its own right such modeling has also created
additional downstream opportunities for the application of more
traditional modeling techniques. Specifically, the ability to
quickly and accurately process high definition images of biopsies
into cell level classification and location data has allowed for
the development and application of more traditional spatial
statistical methods to the resulting data. Despite this, there has been limited work to
build off of the cell classification abilities of modern machine
learning methods. To the author’s knowledge, the most notable
attempts to do so have been Li {.}’s work in 2019,
which utilized data classified by the ConvPath pipeline
(<span class="citation">Wang et al. (<a href="#ref-Wang19" role="doc-biblioref">2019</a>)</span>).</p>
<p>There is ample prior reason to suspect that such applications may
be valuable. <span class="citation">Pagès et al. (<a href="#ref-Pages10" role="doc-biblioref">2010</a>)</span> proposed immune reaction as the seventh hallmark of cancer, and laid out different associations between various types of immune cells and outcomes of interest. Of the different types of immune cells, the most well-studied are lymphocytes.
Lymphocyte infiltration is both
a meaningful prognostic indicator that can help inform treatment
and predict survival across different types of cancer, including
in colorectal cancer (<span class="citation">Idos et al. (<a href="#ref-Idos20" role="doc-biblioref">2020</a>)</span>), Breast cancer
(<span class="citation">Dieci et al. (<a href="#ref-Dieci18" role="doc-biblioref">2018</a>)</span>; <span class="citation">Denkert et al. (<a href="#ref-Denkert15" role="doc-biblioref">2015</a>)</span>), and melanoma
(<span class="citation">Fu et al. (<a href="#ref-Fu19" role="doc-biblioref">2019</a>)</span>). This has led to increased interest in Computational
TILs Assessment (CTA), a sub-field of digital pathology devoted to
developing computational methods to assess lymphocyte infiltration
in biopsy images. Thus far, results that have emerged from CTA
research have provided additional evidence regarding the importance of
tumor infiltrating lymphocytes and their spatial
features in the assessment of pathology images.
<span class="citation">Saltz et al. (<a href="#ref-Saltz18" role="doc-biblioref">2018</a>)</span> specifically
examined the presence of lymphocytes across biopsies in
several types of cancer, and found that certain summarization
metrics of lymphocyte clusters were significantly associated
with survival in certain types of cancer. <span class="citation">Lu et al. (<a href="#ref-Lu20" role="doc-biblioref">2020</a>)</span> also
found that various spatial statistical features of TILs present
in a sample of breast cancer biopsies were significantly associated
with survival both marginally and after adjusting for other factors.
Further, they found that these associations actually differed by
tumor subtype. In addition to this, they found that certain
spatial features of TILs were also significantly associated with
gene expression data.</p>
<p>While the presence of lymphocyte infiltration is consistently associated with positive prognosis and survival outcomes, the same cannot be said for all types of immune cells. It is well accepted that the presence of Tumor-Associated Macrophages (TAMs) generally have a negative impact on survival due to their association with inflammation and their inhibition of adaptive immune response (<span class="citation">Sica, Allavena, and Mantovani (<a href="#ref-Sica08" role="doc-biblioref">2008</a>)</span>, <span class="citation">Tsutsui et al. (<a href="#ref-Tsutsui05" role="doc-biblioref">2005</a>)</span>). While TAMs are less studied in the domain of digital pathology, there has been recent work on the development of pipelines for the classification of TAMs. This work combined with the established effects of TAMs on cancer prognosis suggest both a possible avenue of investigation as well as a method by which to do so.</p>
<p>While this is only a sample of the different types of immune cells that may be present in the Tumor Microenvironment (TME) for a given patient, this broadly suggests that investigating the spatial association between general immune infiltration and the tumor itself may prove fruitful in better understanding the prognostic implications of tumor immunogenicity. Traditionally, immunogenic features of cancer have been assessed (and therfore investigated) at the biopsy level by pathologists. While this macro-assessment allows for assessment of inter-patient heterogeneity, by definition it does not allow for the investigation of -patient heterogeneity. Heterogeneity in the TME, including spatial heterogeneity, has proven to be an important component of understanding cancers of various kinds (<span class="citation">Heindl, Nawaz, and Yuan (<a href="#ref-Heindl15" role="doc-biblioref">2015</a>)</span>; <span class="citation">Yuan (<a href="#ref-Yuan16" role="doc-biblioref">2016</a>)</span>; <span class="citation">Bareche et al. (<a href="#ref-Bareche20" role="doc-biblioref">2020</a>)</span>; <span class="citation">Boxberg et al. (<a href="#ref-Boxberg19" role="doc-biblioref">2019</a>)</span>; <span class="citation">Yan et al. (<a href="#ref-Yan19" role="doc-biblioref">2019</a>)</span>; <span class="citation">Hunter et al. (<a href="#ref-Hunter21" role="doc-biblioref">2021</a>)</span>). It is therefore plausible that developing methods to rigorously model the spatial variation in immune activation will yield insights into both the patient level clinical implications of such heterogeneity as well as possible genomic associations.</p>
<p>Spatial point processes have long been used in the domain of
ecology to rigorously investigate spatial relationships between
various organisms (<span class="citation">Högmander and Särkkä (<a href="#ref-Hogsar99" role="doc-biblioref">1999</a>)</span>; <span class="citation">King et al. (<a href="#ref-King12" role="doc-biblioref">2012</a>)</span>). More
recently, methods from this paradigm have been successfully
applied within the domain of Biostatistics (<span class="citation">Kang et al. (<a href="#ref-Kang11" role="doc-biblioref">2011</a>)</span>;
<span class="citation">Ray, Kang, and Zhang (<a href="#ref-Ray15" role="doc-biblioref">2015</a>)</span>). Given the relatively simple and granular nature
of cell-level spatial imaging data, marked point processes are a
natural way to simultaneously model the randomness in both the
quantities and locations of the different cell types along with
their relative spatial associations.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:01"></span>
<img src="Figures/pipeline_and_cartoon.png" alt="An illustration of spatial interaction between cells of different types and an overview of the pre-processing pipeline. 1a) depicts different strengths of spatial interaction. (A) depicts low/no interaction; (B) depicts moderate interaction; (C) depicts strong interaction. 1b) shows an overview of the SPARTIN pipeline, from pre-processing to biopsy level assessment. Raw biopsy images (A) are pre-processed into cell types (tumor cells or immune cells, colored black and red respectively) and locations. The points are intensity thresholded, removing excess white space from the image and producing a more closely fitted window (B). Next, the biopsy is tessellated into non-overlapping sub-regions containing comparable numbers of tumor cells (C). Models are fit on each tile (D), and tile-specific Cell Type Interaction Probability is calculated from the resulting posterior distributions (E), allowing us to capture heterogeneity in infiltration across the biopsy in a manner that can be aggregated (F) and appropriately summarized (G)." width="100%" />
<p class="caption">
Figure 9.1: An illustration of spatial interaction between cells of different types and an overview of the pre-processing pipeline. 1a) depicts different strengths of spatial interaction. (A) depicts low/no interaction; (B) depicts moderate interaction; (C) depicts strong interaction. 1b) shows an overview of the SPARTIN pipeline, from pre-processing to biopsy level assessment. Raw biopsy images (A) are pre-processed into cell types (tumor cells or immune cells, colored black and red respectively) and locations. The points are intensity thresholded, removing excess white space from the image and producing a more closely fitted window (B). Next, the biopsy is tessellated into non-overlapping sub-regions containing comparable numbers of tumor cells (C). Models are fit on each tile (D), and tile-specific Cell Type Interaction Probability is calculated from the resulting posterior distributions (E), allowing us to capture heterogeneity in infiltration across the biopsy in a manner that can be aggregated (F) and appropriately summarized (G).
</p>
</div>
<p>In this paper, we introduce the SPARTIN (SPatial Analysis of
paRtitioned Tumor-Immune imagiNg) pipeline. SPARTIN is unique
in the spatial pathological imaging analysis literature in that it
uses statistical models to assess the association between tumor
cells and immune cells across an entire partitioned biopsy. This
allows for rigorous quantification of uncertainty in the style of
<span class="citation">Q. Li et al. (<a href="#ref-LiMIM19" role="doc-biblioref">2019</a>)</span> while still allowing for the assessment of full
images in the style of <span class="citation">Saltz et al. (<a href="#ref-Saltz18" role="doc-biblioref">2018</a>)</span>. We accomplish this by
partitioning the cell-level imaging data of each biopsy into
non-overlapping sub-regions, which can then be modeled to capture
the local infiltration patterns using standard techniques from
point process theory.
In section 2, we discuss the theory behind
the models we chose to apply and the details of how the SPARTIN
pipeline works. We also define Cell Type Infiltration Probability (CTIP), a measure of local immune cell infiltration. In section 3, we present results from a simulation
study as well as an application to an actual data set consisting of
SKCM biopsies.
Finally, in section 4, we discuss the implications
of this pipeline as well as potential future directions for the
work.</p>
<p>The results of our simulations suggest that CTIP reliably identifies
local positive spatial associations across a variety of different
combinations of cell abundances and strengths of associations. We also found that CTIP was significantly
associated with biopsy-level gene expression among genes related to immune cells. Additionally, we found that CTIP was significantly associated with various outcomes as assessed by <span class="citation">Akbani et al. (<a href="#ref-Akbani15" role="doc-biblioref">2015</a>)</span>, including transcriptomic class and pathologist assessment.
Finally, at the patient outcome level, we found that immune cell infiltration as
measured by CTIP was significantly associated
with overall survival.</p>
<p>We focus our analysis on Skin Cutaneous Melanoma
(SKCM). SKCM is an appealing target for the investigation of
immune cell infiltration for several reasons.
SKCM has been shown
to be particularly responsive to Immunotherapy in some cases
(<span class="citation">Franklin et al. (<a href="#ref-Franklin17" role="doc-biblioref">2017</a>)</span>; <span class="citation">Achkar and Tarhini (<a href="#ref-Achkar17" role="doc-biblioref">2017</a>)</span>). It is possible that
the ability to quantify immune infiltration at a large scale
may allow for more detailed investigation into the scenarios in
which this treatment may be most effectively deployed. It is also
well established that SKCM has an unusually high mutational load
amongst the various cancer types (<span class="citation">Berger et al. (<a href="#ref-Berger12" role="doc-biblioref">2012</a>)</span>). The
ability to quantify infiltration may allow for further
investigation into not only genomic associations of this<br />
occurrence but associations with mutations as well.</p>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Achkar17" class="csl-entry">
Achkar, Tala, and Ahmad A Tarhini. 2017. <span>“The Use of Immunotherapy in the Treatment of Melanoma.”</span> <em>Journal of Hematology &amp; Oncology</em> 10 (1): 1–9.
</div>
<div id="ref-Akbani15" class="csl-entry">
Akbani, Rehan, Kadir C Akdemir, B Arman Aksoy, Monique Albert, Adrian Ally, Samirkumar B Amin, Harindra Arachchi, et al. 2015. <span>“Genomic Classification of Cutaneous Melanoma.”</span> <em>Cell</em> 161 (7): 1681–96.
</div>
<div id="ref-Amgad19" class="csl-entry">
Amgad, Mohamed, Anindya Sarkar, Chukka Srinivas, Rachel Redman, Simrath Ratra, Charles J Bechert, Benjamin C Calhoun, et al. 2019. <span>“Joint Region and Nucleus Segmentation for Characterization of Tumor Infiltrating Lymphocytes in Breast Cancer.”</span> In <em>Medical Imaging 2019: Digital Pathology</em>, 10956:129–36. SPIE.
</div>
<div id="ref-Bareche20" class="csl-entry">
Bareche, Yacine, Laurence Buisseret, Tina Gruosso, Edwina Girard, David Venet, Floriane Dupont, Christine Desmedt, et al. 2020. <span>“Unraveling Triple-Negative Breast Cancer Tumor Microenvironment Heterogeneity: Towards an Optimized Treatment Approach.”</span> <em>JNCI: Journal of the National Cancer Institute</em> 112 (7): 708–19.
</div>
<div id="ref-Berger12" class="csl-entry">
Berger, Michael F, Eran Hodis, Timothy P Heffernan, Yonathan Lissanu Deribe, Michael S Lawrence, Alexei Protopopov, Elena Ivanova, et al. 2012. <span>“Melanoma Genome Sequencing Reveals Frequent Prex2 Mutations.”</span> <em>Nature</em> 485 (7399): 502–6.
</div>
<div id="ref-Bian21" class="csl-entry">
Bian, Yun, Yan Fang Liu, Hui Jiang, Yinghao Meng, Fang Liu, Kai Cao, Hao Zhang, et al. 2021. <span>“Machine Learning for MRI Radiomics: A Study Predicting Tumor-Infiltrating Lymphocytes in Patients with Pancreatic Ductal Adenocarcinoma.”</span> <em>Abdominal Radiology</em> 46 (10): 4800–4816.
</div>
<div id="ref-Boxberg19" class="csl-entry">
Boxberg, Melanie, Lena Leising, Katja Steiger, Moritz Jesinghaus, Aezlat Alkhamas, Marion Mielke, Nicole Pfarr, et al. 2019. <span>“Composition and Clinical Impact of the Immunologic Tumor Microenvironment in Oral Squamous Cell Carcinoma.”</span> <em>The Journal of Immunology</em> 202 (1): 278–91.
</div>
<div id="ref-Denkert15" class="csl-entry">
Denkert, Carsten, Gunter Von Minckwitz, Jan C Brase, Bruno V Sinn, Stephan Gade, Ralf Kronenwett, Berit M Pfitzner, et al. 2015. <span>“Tumor-Infiltrating Lymphocytes and Response to Neoadjuvant Chemotherapy with or Without Carboplatin in Human Epidermal Growth Factor Receptor 2-Positive and Triple-Negative Primary Breast Cancers.”</span> <em>J Clin Oncol</em> 33 (9): 983–91.
</div>
<div id="ref-Dieci18" class="csl-entry">
Dieci, Maria Vittoria, Nina Radosevic-Robin, Susan Fineberg, Gert Van den Eynden, Nils Ternes, Frédérique Penault-Llorca, Giancarlo Pruneri, et al. 2018. <span>“Update on Tumor-Infiltrating Lymphocytes (TILs) in Breast Cancer, Including Recommendations to Assess TILs in Residual Disease After Neoadjuvant Therapy and in Carcinoma in Situ: A Report of the International Immuno-Oncology Biomarker Working Group on Breast Cancer.”</span> In <em>Seminars in Cancer Biology</em>, 52:16–25. Elsevier.
</div>
<div id="ref-Franklin17" class="csl-entry">
Franklin, C, E Livingstone, A Roesch, B Schilling, and D Schadendorf. 2017. <span>“Immunotherapy in Melanoma: Recent Advances and Future Directions.”</span> <em>European Journal of Surgical Oncology (EJSO)</em> 43 (3): 604–11.
</div>
<div id="ref-Fu19" class="csl-entry">
Fu, Qiaofen, Nan Chen, Chunlei Ge, Ruilei Li, Zhen Li, Baozhen Zeng, Chunyan Li, et al. 2019. <span>“Prognostic Value of Tumor-Infiltrating Lymphocytes in Melanoma: A Systematic Review and Meta-Analysis.”</span> <em>Oncoimmunology</em> 8 (7): e1593806.
</div>
<div id="ref-Heindl15" class="csl-entry">
Heindl, Andreas, Sidra Nawaz, and Yinyin Yuan. 2015. <span>“Mapping Spatial Heterogeneity in the Tumor Microenvironment: A New Era for Digital Pathology.”</span> <em>Laboratory Investigation</em> 95 (4): 377–84.
</div>
<div id="ref-Hogsar99" class="csl-entry">
Högmander, Harri, and Aila Särkkä. 1999. <span>“Multitype Spatial Point Patterns with Hierarchical Interactions.”</span> <em>Biometrics</em> 55 (4): 1051–58.
</div>
<div id="ref-Hunter21" class="csl-entry">
Hunter, Miranda V, Reuben Moncada, Joshua M Weiss, Itai Yanai, and Richard M White. 2021. <span>“Spatially Resolved Transcriptomics Reveals the Architecture of the Tumor-Microenvironment Interface.”</span> <em>Nature Communications</em> 12 (1): 1–16.
</div>
<div id="ref-Idos20" class="csl-entry">
Idos, Gregory E, Janet Kwok, Nirupama Bonthala, Lynn Kysh, Stephen B Gruber, and Chenxu Qu. 2020. <span>“The Prognostic Implications of Tumor Infiltrating Lymphocytes in Colorectal Cancer: A Systematic Review and Meta-Analysis.”</span> <em>Scientific Reports</em> 10 (1): 1–14.
</div>
<div id="ref-Kang11" class="csl-entry">
Kang, Jian, Timothy D Johnson, Thomas E Nichols, and Tor D Wager. 2011. <span>“Meta Analysis of Functional Neuroimaging Data via Bayesian Spatial Point Processes.”</span> <em>Journal of the American Statistical Association</em> 106 (493): 124–34.
</div>
<div id="ref-King12" class="csl-entry">
King, Ruth, Janine B Illian, Stuart E King, Glenna F Nightingale, and Ditte K Hendrichsen. 2012. <span>“A Bayesian Approach to Fitting Gibbs Processes with Temporal Random Effects.”</span> <em>Journal of Agricultural, Biological, and Environmental Statistics</em> 17 (4): 601–22.
</div>
<div id="ref-Komura18" class="csl-entry">
Komura, Daisuke, and Shumpei Ishikawa. 2018. <span>“Machine Learning Methods for Histopathological Image Analysis.”</span> <em>Computational and Structural Biotechnology Journal</em> 16: 34–42.
</div>
<div id="ref-LiMIM19" class="csl-entry">
Li, Qiwei, Xinlei Wang, Faming Liang, and Guanghua Xiao. 2019. <span>“A Bayesian Mark Interaction Model for Analysis of Tumor Pathology Images.”</span> <em>The Annals of Applied Statistics</em> 13 (3): 1708.
</div>
<div id="ref-Lu20" class="csl-entry">
Lu, Zixiao, Siwen Xu, Wei Shao, Yi Wu, Jie Zhang, Zhi Han, Qianjin Feng, and Kun Huang. 2020. <span>“Deep-Learning–Based Characterization of Tumor-Infiltrating Lymphocytes in Breast Cancers from Histopathology Images and Multiomics Data.”</span> <em>JCO Clinical Cancer Informatics</em> 4: 480–90.
</div>
<div id="ref-Negahbani21" class="csl-entry">
Negahbani, Farzin, Rasool Sabzi, Bita Pakniyat Jahromi, Dena Firouzabadi, Fateme Movahedi, Mahsa Kohandel Shirazi, Shayan Majidi, and Amirreza Dehghanian. 2021. <span>“PathoNet Introduced as a Deep Neural Network Backend for Evaluation of Ki-67 and Tumor-Infiltrating Lymphocytes in Breast Cancer.”</span> <em>Scientific Reports</em> 11 (1): 1–13.
</div>
<div id="ref-Pages10" class="csl-entry">
Pagès, F, J Galon, M-C Dieu-Nosjean, E Tartour, C Sautès-Fridman, and W-H Fridman. 2010. <span>“Immune Infiltration in Human Tumors: A Prognostic Factor That Should Not Be Ignored.”</span> <em>Oncogene</em> 29 (8): 1093–1102. <a href="https://doi.org/10.1038/onc.2009.416">https://doi.org/10.1038/onc.2009.416</a>.
</div>
<div id="ref-Ray15" class="csl-entry">
Ray, Meredith, Jian Kang, and Hongmei Zhang. 2015. <span>“Identifying Activation Centers with Spatial Cox Point Processes Using Fmri Data.”</span> <em>IEEE/ACM Transactions on Computational Biology and Bioinformatics</em> 13 (6): 1130–41.
</div>
<div id="ref-Saltz18" class="csl-entry">
Saltz, Joel, Rajarsi Gupta, Le Hou, Tahsin Kurc, Pankaj Singh, Vu Nguyen, Dimitris Samaras, et al. 2018. <span>“Spatial Organization and Molecular Correlation of Tumor-Infiltrating Lymphocytes Using Deep Learning on Pathology Images.”</span> <em>Cell Reports</em> 23 (1): 181–93.
</div>
<div id="ref-Sica08" class="csl-entry">
Sica, Antonio, Paola Allavena, and Alberto Mantovani. 2008. <span>“Cancer Related Inflammation: The Macrophage Connection.”</span> <em>Cancer Letters</em> 267 (2): 204–15. <a href="https://doi.org/10.1016/j.canlet.2008.03.028">https://doi.org/10.1016/j.canlet.2008.03.028</a>.
</div>
<div id="ref-Tsutsui05" class="csl-entry">
Tsutsui, Shinichi, Kazuhiro Yasuda, Kosuke Suzuki, Kouichirou Tahara, Hidefumi Higashi, and Shoichi Era. 2005. <span>“Macrophage Infiltration and Its Prognostic Implications in Breast Cancer: The Relationship with VEGF Expression and Microvessel Density.”</span> <em>Oncology Reports</em>, August. <a href="https://doi.org/10.3892/or.14.2.425">https://doi.org/10.3892/or.14.2.425</a>.
</div>
<div id="ref-Wang19" class="csl-entry">
Wang, Shidan, Tao Wang, Lin Yang, Donghan M Yang, Junya Fujimoto, Faliu Yi, Xin Luo, et al. 2019. <span>“ConvPath: A Software Tool for Lung Adenocarcinoma Digital Pathological Image Analysis Aided by a Convolutional Neural Network.”</span> <em>EBioMedicine</em> 50: 103–10.
</div>
<div id="ref-Yan19" class="csl-entry">
Yan, Yiyi, Alexey A Leontovich, Michael J Gerdes, Keyur Desai, Jinhong Dong, Anup Sood, Alberto Santamaria-Pang, et al. 2019. <span>“Understanding Heterogeneous Tumor Microenvironment in Metastatic Melanoma.”</span> <em>PloS One</em> 14 (6): e0216485.
</div>
<div id="ref-Yuan16" class="csl-entry">
Yuan, Yinyin. 2016. <span>“Spatial Heterogeneity in the Tumor Microenvironment.”</span> <em>Cold Spring Harbor Perspectives in Medicine</em> 6 (8): a026583.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="preface-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="materials-and-methods.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/scratch.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
