<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>S.10 Materials and Methods | SPARTIN Supplementary material</title>
  <meta name="description" content="Supplementary material related to the SPARTYN pipeline" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="S.10 Materials and Methods | SPARTIN Supplementary material" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Supplementary material related to the SPARTYN pipeline" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="S.10 Materials and Methods | SPARTIN Supplementary material" />
  
  <meta name="twitter:description" content="Supplementary material related to the SPARTYN pipeline" />
  

<meta name="author" content="Nate Osher" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction.html"/>
<link rel="next" href="simulation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="model-fitting.html"><a href="model-fitting.html"><i class="fa fa-check"></i><b>2</b> Model Fitting</a></li>
<li class="chapter" data-level="3" data-path="biopsy-partition.html"><a href="biopsy-partition.html"><i class="fa fa-check"></i><b>3</b> Biopsy Partition</a></li>
<li class="chapter" data-level="4" data-path="cell-classification-model.html"><a href="cell-classification-model.html"><i class="fa fa-check"></i><b>4</b> Cell Classification Model</a>
<ul>
<li class="chapter" data-level="4.1" data-path="cell-classification-model.html"><a href="cell-classification-model.html#whole-slide-image-wsi-retrieval-and-obtaining-training-labels"><i class="fa fa-check"></i><b>4.1</b> Whole-Slide Image (WSI) Retrieval and Obtaining Training Labels</a></li>
<li class="chapter" data-level="4.2" data-path="cell-classification-model.html"><a href="cell-classification-model.html#whole-slide-image-wsi-nucleus-segmentation"><i class="fa fa-check"></i><b>4.2</b> Whole-Slide Image (WSI) Nucleus Segmentation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genomic-associations.html"><a href="genomic-associations.html"><i class="fa fa-check"></i><b>5</b> Genomic Associations</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genomic-associations.html"><a href="genomic-associations.html#association-with-expression-of-significantly-mutated-genes"><i class="fa fa-check"></i><b>5.1</b> Association with expression of significantly mutated genes</a></li>
<li class="chapter" data-level="5.2" data-path="genomic-associations.html"><a href="genomic-associations.html#association-with-expression-of-immune-genes"><i class="fa fa-check"></i><b>5.2</b> Association with expression of immune genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="association-with-deconvolution-data.html"><a href="association-with-deconvolution-data.html"><i class="fa fa-check"></i><b>6</b> Association with Deconvolution Data</a></li>
<li class="chapter" data-level="7" data-path="spartin-r-package.html"><a href="spartin-r-package.html"><i class="fa fa-check"></i><b>7</b> SPARTIN R package</a></li>
<li class="chapter" data-level="8" data-path="preface-1.html"><a href="preface-1.html"><i class="fa fa-check"></i><b>8</b> Preface</a></li>
<li class="chapter" data-level="9" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>9</b> Introduction</a></li>
<li class="chapter" data-level="10" data-path="materials-and-methods.html"><a href="materials-and-methods.html"><i class="fa fa-check"></i><b>10</b> Materials and Methods</a>
<ul>
<li class="chapter" data-level="10.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#overview"><i class="fa fa-check"></i><b>10.1</b> Overview</a></li>
<li class="chapter" data-level="10.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#spatial-point-process-models"><i class="fa fa-check"></i><b>10.2</b> Spatial Point Process Models</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#data-structure"><i class="fa fa-check"></i><b>10.2.1</b> Data Structure</a></li>
<li class="chapter" data-level="10.2.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#model"><i class="fa fa-check"></i><b>10.2.2</b> Model</a></li>
<li class="chapter" data-level="10.2.3" data-path="materials-and-methods.html"><a href="materials-and-methods.html#interpretation"><i class="fa fa-check"></i><b>10.2.3</b> Interpretation</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="materials-and-methods.html"><a href="materials-and-methods.html#pseudolikelihood-function"><i class="fa fa-check"></i><b>10.3</b> Pseudolikelihood Function</a></li>
<li class="chapter" data-level="10.4" data-path="materials-and-methods.html"><a href="materials-and-methods.html#intensity-thresholding-and-partition"><i class="fa fa-check"></i><b>10.4</b> Intensity Thresholding and Partition</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#motivation-for-partition"><i class="fa fa-check"></i><b>10.4.1</b> Motivation for partition</a></li>
<li class="chapter" data-level="10.4.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#partition-pipeline"><i class="fa fa-check"></i><b>10.4.2</b> Partition pipeline</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="materials-and-methods.html"><a href="materials-and-methods.html#cell-type-interaction-probability"><i class="fa fa-check"></i><b>10.5</b> Cell Type Interaction Probability</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#motivation"><i class="fa fa-check"></i><b>10.5.1</b> Motivation</a></li>
<li class="chapter" data-level="10.5.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#definition"><i class="fa fa-check"></i><b>10.5.2</b> Definition</a></li>
<li class="chapter" data-level="10.5.3" data-path="materials-and-methods.html"><a href="materials-and-methods.html#estimating-the-empirical-null-distribution"><i class="fa fa-check"></i><b>10.5.3</b> Estimating the empirical null distribution</a></li>
<li class="chapter" data-level="10.5.4" data-path="materials-and-methods.html"><a href="materials-and-methods.html#computation-of-ctip"><i class="fa fa-check"></i><b>10.5.4</b> Computation of CTIP</a></li>
<li class="chapter" data-level="10.5.5" data-path="materials-and-methods.html"><a href="materials-and-methods.html#interpretation-1"><i class="fa fa-check"></i><b>10.5.5</b> Interpretation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="simulation.html"><a href="simulation.html"><i class="fa fa-check"></i><b>11</b> Simulation</a>
<ul>
<li class="chapter" data-level="11.1" data-path="simulation.html"><a href="simulation.html#simulation-overview"><i class="fa fa-check"></i><b>11.1</b> Simulation Overview</a></li>
<li class="chapter" data-level="11.2" data-path="simulation.html"><a href="simulation.html#negative-simulations"><i class="fa fa-check"></i><b>11.2</b> Negative simulations</a></li>
<li class="chapter" data-level="11.3" data-path="simulation.html"><a href="simulation.html#positive-simulations"><i class="fa fa-check"></i><b>11.3</b> Positive simulations</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="application.html"><a href="application.html"><i class="fa fa-check"></i><b>12</b> Application</a>
<ul>
<li class="chapter" data-level="12.1" data-path="application.html"><a href="application.html#background"><i class="fa fa-check"></i><b>12.1</b> Background</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="application.html"><a href="application.html#data-and-pre-processing"><i class="fa fa-check"></i><b>12.1.1</b> Data and Pre-Processing</a></li>
<li class="chapter" data-level="12.1.2" data-path="application.html"><a href="application.html#cell-classification-model-1"><i class="fa fa-check"></i><b>12.1.2</b> Cell Classification Model</a></li>
<li class="chapter" data-level="12.1.3" data-path="application.html"><a href="application.html#melanoma"><i class="fa fa-check"></i><b>12.1.3</b> Melanoma</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="application.html"><a href="application.html#genomic-associations-1"><i class="fa fa-check"></i><b>12.2</b> Genomic Associations</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="application.html"><a href="application.html#association-with-expression-of-significantly-mutated-genes-1"><i class="fa fa-check"></i><b>12.2.1</b> Association with expression of significantly mutated genes</a></li>
<li class="chapter" data-level="12.2.2" data-path="application.html"><a href="application.html#association-with-expression-of-immune-genes-1"><i class="fa fa-check"></i><b>12.2.2</b> Association with expression of immune genes</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="application.html"><a href="application.html#association-with-deconvolution-data-1"><i class="fa fa-check"></i><b>12.3</b> Association with Deconvolution Data</a></li>
<li class="chapter" data-level="12.4" data-path="application.html"><a href="application.html#association-with-other-outcomes-of-interest"><i class="fa fa-check"></i><b>12.4</b> Association with Other Outcomes of Interest</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="application.html"><a href="application.html#transcriptomic-classes"><i class="fa fa-check"></i><b>12.4.1</b> Transcriptomic classes</a></li>
<li class="chapter" data-level="12.4.2" data-path="application.html"><a href="application.html#pathologist-assessment"><i class="fa fa-check"></i><b>12.4.2</b> Pathologist assessment</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="application.html"><a href="application.html#survival-analysis"><i class="fa fa-check"></i><b>12.5</b> Survival Analysis</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="discussion-and-limitations.html"><a href="discussion-and-limitations.html"><i class="fa fa-check"></i><b>13</b> Discussion and Limitations</a>
<ul>
<li class="chapter" data-level="13.1" data-path="discussion-and-limitations.html"><a href="discussion-and-limitations.html#r-markdown"><i class="fa fa-check"></i><b>13.1</b> R Markdown</a></li>
<li class="chapter" data-level="13.2" data-path="discussion-and-limitations.html"><a href="discussion-and-limitations.html#including-plots"><i class="fa fa-check"></i><b>13.2</b> Including Plots</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">SPARTIN Supplementary material</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="materials-and-methods" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">S.10</span> Materials and Methods<a href="materials-and-methods.html#materials-and-methods" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="overview" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Overview<a href="materials-and-methods.html#overview" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Figure <a href="introduction.html#fig:01">9.1</a>b) illustrates the analysis pipeline for the processed data. For each
biopsy, the smallest window that fit all cells in the biopsy is constructed by thresholding the image intensity
(Figure 1, step A to step B).
This window is then divided into tiles that fully partition it,
while containing similar numbers of tumor cells (step B to step C). A Bayesian spatial
point process model is then fit on each of these tiles separately,
yielding a posterior distribution of the local interaction parameter (step C to step D).
For each tile, this local distribution is compared to a
tile-specific null distribution to compute the local CTIP value (step E), before being combined to summarize
the overall level of interaction at the biopsy level (steps F and G).</p>
</div>
<div id="spatial-point-process-models" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Spatial Point Process Models<a href="materials-and-methods.html#spatial-point-process-models" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data-structure" class="section level3 hasAnchor" number="10.2.1">
<h3><span class="header-section-number">10.2.1</span> Data Structure<a href="materials-and-methods.html#data-structure" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let <span class="math inline">\(B_1...B_n\)</span> be a set of independent regions with <span class="math inline">\(B_i \in \mathbb{R}^2 \times \mathbb{R}^2\)</span>, such that each <span class="math inline">\(B_i\)</span> has been partitioned into <span class="math inline">\(n_i\)</span> non-overlapping sub-regions <span class="math inline">\(b_{i1}..b_{in_i}\)</span> each with well-defined boundaries. Denote the set of points observed within sub-region <span class="math inline">\(j\)</span> of region <span class="math inline">\(i\)</span> by <span class="math inline">\({\mathbf{x}}_{ij}\)</span>, and the marks of those points as <span class="math inline">\({\mathbf{m}}_{ij}\)</span> where each mark <span class="math inline">\(m_{ijk} \in \{K_1,...,K_\ell\}\)</span>. Within each sub-region, this data can be naturally viewed as the realization of a marked point process and modeled as a Hierarchical Multitype Strauss Process.</p>
</div>
<div id="model" class="section level3 hasAnchor" number="10.2.2">
<h3><span class="header-section-number">10.2.2</span> Model<a href="materials-and-methods.html#model" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The usage of a Strauss model to study spatial interaction in a marked point process is natural, since Strauss models were
originally conceived for this purpose
(<span class="citation">Strauss (<a href="#ref-Strauss75" role="doc-biblioref">1975</a>)</span>) . As for the specific decision to use
the hierarchical variant over the more standard multitype model,
there were two major considerations that influenced our decision.
First, in our application treating the locations of the immune cells as conditional
upon the locations of the tumor cells is a priori biologically
plausible. Moreover, it is quite plausible that while
immune cells are responsive to the positioning of tumor cells, tumor
cells may not be similarly responsive to the positioning of
immune cells. The hierarchical Strauss models reflects the
plausibly unidirectional nature of this spatial relationship in its
modeling assumptions. Second, the hierarchical variant of the
Strauss model allows for the proper modeling of positive
interaction between points of different types, while the standard
multitype model does not. Because we are interested in modeling not only negative interaction between different cell types but positive interaction as well, the hierarchical variant is the clear choice. It bears mentioning that this model
can be extended to arbitrary numbers of marks. For our current
application, we have limited ourselves to two, but it would
be entirely possible to extend the chosen model to as many
cell types as were available for study in the future.</p>
<p>The density function of the Hierarchical Multitype Strauss Process when there are <span class="math inline">\(\ell = 2\)</span> qualitative marks
is defined by</p>
<p><span class="math display">\[\begin{equation}
\label{eqn:density}
f({\mathbf{p}}_1, {\mathbf{p}}_2) \propto
\beta_1^{n_1} \beta_2^{n_2}
\gamma_{11}^{S_{R_{11}}({\mathbf{p}}_1)}
\gamma_{22}^{S_{R_{2}}({\mathbf{p}}_2)}
\gamma_{12}^{S_{R_{12}}({\mathbf{p}}_1, {\mathbf{p}}_2)}
\tag{1}  
\end{equation}\]</span></p>
<p>where <span class="math inline">\({\mathbf{p}}_t\)</span> is a vector of points of type <span class="math inline">\(t\)</span>, <span class="math inline">\(n_t\)</span> is the number of points of type <span class="math inline">\(t\)</span>, <span class="math inline">\(\beta_t\)</span> is the first order intensity of points of type <span class="math inline">\(t\)</span>, <span class="math inline">\(S_{R_{tl}}(\cdot)\)</span> counts the number of pairs of points of types <span class="math inline">\(t\)</span> and <span class="math inline">\(l\)</span> within <span class="math inline">\(R_{tl}\)</span> of one another where <span class="math inline">\(R_{tl}\)</span> is selected a priori based on subject specific knowledge, and <span class="math inline">\(\gamma_{tl}\)</span> captures the tendency of points of type <span class="math inline">\(l\)</span> to be near points of type <span class="math inline">\(t\)</span>.</p>
</div>
<div id="interpretation" class="section level3 hasAnchor" number="10.2.3">
<h3><span class="header-section-number">10.2.3</span> Interpretation<a href="materials-and-methods.html#interpretation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As previously mentioned, <span class="math inline">\(\gamma_{tl}\)</span> can be thought of as the
degree to which points of type <span class="math inline">\(l\)</span> tend to be close to or far away
from points of type <span class="math inline">\(t\)</span>. This allows us to distinguish between
the mere relative abundance of different types of points (which is
captured by the <span class="math inline">\(\beta_1\)</span> and <span class="math inline">\(\beta_2\)</span> parameters) and the
actual spatial associations between the different types of points.
This distinction is important in situations where there are
substantial numbers of points of both types but no positive
spatial association (and possibly a negative one). For examples,
see section 3 for results from the simulation study. When
<span class="math inline">\(t = 1\)</span> and <span class="math inline">\(l = 2\)</span>, this association
is interpreted as conditional upon the locations of the type 1
points. Under the Hierarchical Strauss Model,
<span class="math inline">\(\gamma_{12} \in [0, \infty)\)</span>, with <span class="math inline">\(\gamma_{12} \in [0,1)\)</span>
implying negative interaction, <span class="math inline">\(\gamma_{12} \in (1, \infty)\)</span>
implying negative interaction, and <span class="math inline">\(\gamma_{12} = 1\)</span> implying
no interaction at all. As one might intuitively expect, larger values of
<span class="math inline">\(\gamma_{12}\)</span> correspond to more positive interaction, and
smaller values correspond to a more negative interaction. Due to mathematical constraints on the density,
we must have <span class="math inline">\(\gamma_{11}, \gamma_{22} \in [0,1]\)</span>. This
constrains interaction between points of the same
type to be modeled as negative. Since this constraint is unlikely
to be satisfied in the context of a tumor biopsy, we set
<span class="math inline">\(\gamma_{11} = \gamma_{12} = 1\)</span>, in effect assuming no
interaction between cells of the same type. Based on prior
biological knowledge, we set <span class="math inline">\(R_{12} = 30\)</span> <span class="math inline">\(\mu m\)</span>.</p>
<p>In practice for various reasons these parameters are modeled on
the log-scale. While this alters interpretation,
because the <span class="math inline">\(log(\cdot)\)</span> function is a monotonically increasing
function, the basic intuition still holds in that larger values
on the log-scale are still indicative of stronger spatial
interaction. An additional advantage of modeling on the log scale
is the relative ease of interpretability. Whereas on the normal
scale values of <span class="math inline">\(\gamma_{12}\)</span> between 0 and 1 are indicative of
negative interaction, values between 1 and <span class="math inline">\(\infty\)</span> are
indicative of positive interaction, and 1 is indicative of no interaction,
on the log scale negative values of <span class="math inline">\(\log(\gamma_{12})\)</span> indicate negative
interaction, positive values indicate positive interaction, and a value
of 0 indicates no interaction.</p>
</div>
</div>
<div id="pseudolikelihood-function" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Pseudolikelihood Function<a href="materials-and-methods.html#pseudolikelihood-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Except in the trivial case where <span class="math inline">\(\gamma_{12} = 1\)</span>, the normalizing constant of this distribution is computationally intractable. This motivates the usage of the pseudolikelihood as outlined in <span class="citation">Baddeley and Turner (<a href="#ref-BT00" role="doc-biblioref">2000</a>)</span>. For the density <span class="math inline">\(f(\cdot)\)</span> of the simplified Hierarchical Strauss model outlined above, define the conditional intensity function at a point <span class="math inline">\(u\)</span> given <span class="math inline">\(\theta = \{\beta_1, \beta_2, \gamma_{12}\}\)</span> and points <span class="math inline">\({\mathbf{x}}\)</span> observed in window <span class="math inline">\(A\)</span> by</p>
<p><span class="math display">\[\begin{equation}
\label{eqn:CL}
\lambda(u|\theta, {\mathbf{x}}) =
\begin{cases}
\frac{f({\mathbf{x}} \cup \{u\})}{f({\mathbf{x}})} &amp; u \not\in {\mathbf{x}} \\
\frac{f({\mathbf{x}})}{f({\mathbf{x}} - \{u\})} &amp; u \in {\mathbf{x}}
\end{cases}
\tag{2}
\end{equation}\]</span></p>
<p>Given this, the pseudolikelihood is defined by</p>
<p><span class="math display">\[\begin{equation}
\label{eqn:PL}
PL(\theta | {\mathbf{x}}) =
\prod_{x_i \in {\mathbf{x}}} \lambda(x_i | \theta, {\mathbf{x}}) \exp(-\int_A \lambda(u | \theta, {\mathbf{x}}) du) \tag{3}
\end{equation}\]</span></p>
<p>This provides a computationally tractable alternative to the likelihood function that can be used in inference. Specifically, the integral in the pseudolikelihood function can be easily approximated by summing over a weighted quadrature on <span class="math inline">\(A\)</span>. Thus, given a quadrature <span class="math inline">\({\mathbf{u}}\)</span> on <span class="math inline">\(A\)</span> with corresponding weights <span class="math inline">\({\mathbf{w}}\)</span>, Equation (<span class="math inline">\(\ref{eqn:PL}\)</span>) can be approximated by</p>
<p><span class="math display">\[\begin{equation}
\label{eqn:PL_approx}
PL(\theta | {\mathbf{x}}) \approx
\prod_{x_i \in {\mathbf{x}}} \lambda(x_i | \theta, {\mathbf{x}}) \exp(-\sum_{u_j \in {\mathbf{u}}} \lambda(u_j | \theta, {\mathbf{x}}) w_j) \tag{4}
\end{equation}\]</span></p>
<p>The density of the selected quadrature <span class="math inline">\({\mathbf{u}}\)</span> on <span class="math inline">\(A\)</span> is chosen to balance the accuracy of the approximation against the computational load that larger quadratures impose.</p>
<p>Using this approximation of the pseudolikelihood function in place of the more standard likelihood function, Bayesian analysis can proceed in the style of <span class="citation">King et al. (<a href="#ref-King12" role="doc-biblioref">2012</a>)</span> by simply assigning priors to the parameters of interest and using techniques to sample from non-closed form posterior likelihoods. We assigned non-informative normal priors with mean 0 and variance <span class="math inline">\(10^6\)</span> to <span class="math inline">\(\log(\gamma_{12})\)</span>, <span class="math inline">\(\log(\beta_1)\)</span>, and <span class="math inline">\(\log(\beta_2)\)</span>. In all analysis presented below the quadrature and weights used to estimate the integral in the pseudolikelihood function was generated by the spatstat package (<span class="citation">Baddeley and Turner (<a href="#ref-BT05" role="doc-biblioref">2005</a>)</span>). Samples from the posterior were taken using JAGS via the R2jags package (<span class="citation">Su et al. (<a href="#ref-Su15" role="doc-biblioref">2015</a>)</span>). In order to use JAGS to fit the model, the so-called “Poisson zero trick” as described in <span class="citation">Kruschke (<a href="#ref-Kruschke14" role="doc-biblioref">2014</a>)</span>. Essentially, this allows for the fitting of a model with an arbitrary likelihood (or pseudolikelihood) function. Results were checked against both frequentist model fittings from the spatstat <code>ppm</code>/<code>hierstrauss</code> functions as well as STAN model fittings (<span class="citation">Carpenter et al. (<a href="#ref-Carpenter17" role="doc-biblioref">2017</a>)</span>) and yielded virtually indistinguishable results in both cases.</p>
</div>
<div id="intensity-thresholding-and-partition" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> Intensity Thresholding and Partition<a href="materials-and-methods.html#intensity-thresholding-and-partition" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="motivation-for-partition" class="section level3 hasAnchor" number="10.4.1">
<h3><span class="header-section-number">10.4.1</span> Motivation for partition<a href="materials-and-methods.html#motivation-for-partition" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Each biopsy was partitioned into non-overlapping sub-regions,
each of which was modeled separately as a point process. The
primary motivation for structuring the analysis in this manner
came from prior biological knowledge about tumor composition.
Tumors are heterogeneous entities in many respects, and
immune infiltration is no exception. Thus, partitioning
the biopsy into non-overlapping sub-regions and fitting models
on each sub-region is a natural way to capture this heterogeneity.
An additional benefit of this method is that it allows for the
parallelization of model fitting across the resulting sub-regions
within a single biopsy. Much like selecting a quadrature when approximating
an integral, calibrating the fineness of the partition entails a
tradeoff between the precision with which one can assess the
spatial heterogeneity within the biopsy and the computational
load that a finer partition entails.</p>
</div>
<div id="partition-pipeline" class="section level3 hasAnchor" number="10.4.2">
<h3><span class="header-section-number">10.4.2</span> Partition pipeline<a href="materials-and-methods.html#partition-pipeline" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In order to partition a given biopsy into non-overlapping
sub-regions, we began with the smallest bounding rectangular
window that contained cells <span class="math inline">\(1...c_i\)</span>. We then applied an
intensity thresholding algorithm in order to find the
smallest possible window that still contained all <span class="math inline">\(c_i\)</span> cells.
This was accomplished using the <code>density.ppp</code> function from the
<code>spatstat</code> package (<span class="citation">Baddeley and Turner (<a href="#ref-BT05" role="doc-biblioref">2005</a>)</span>). The bounding rectangular window was
broken up into small (15<span class="math inline">\(\mu m\)</span> <span class="math inline">\(\times\)</span> 15<span class="math inline">\(\mu m\)</span>) pixels, the
intensity of which was estimated as a function of the number of
points per square unit of area of the pixel. These values were
also “smoothed” between pixels according to a pre-selected
bandwidth of 25<span class="math inline">\(\mu m\)</span> chosen through experimentation. The
final window was constructed by combining all pixels that were
above a certain threshold into a window. The “pixel” size,
smoothing bandwidth, and intensity threshold collectively
determined the final window, and the same settings (15<span class="math inline">\(\mu m^2\)</span>
pixel size and <span class="math inline">\(25\mu m\)</span> smoothing bandwidth) were applied
across all biopsies.
Next, we applied a voronoi tessellation to the <span class="math inline">\(T_i\)</span> tumor cells
within the intensity thresholded window, partitioning it into
tumor cell specific sub-windows <span class="math inline">\(1...T_i\)</span> corresponding to tumor
cells <span class="math inline">\(1...T_i\)</span>. We then applied a modified version of k-means to
the tumor cells, such that each of the <span class="math inline">\(K_i\)</span> resulting clusters
was constrained to be between a pre-defined range of cell counts.
Finally, each tumor cell specific sub-window within a given
cluster was combined into a tile, corresponding to each of the
<span class="math inline">\(K_i\)</span> clusters from the k-means clustering; see Figure 1 for
an illustration of the process. This results in <span class="math inline">\(K_i\)</span>
tiles that fully partition the intensity thresholded window. This
partition uniquely defines the membership of each of the <span class="math inline">\(c_i\)</span>
total cells (tumor or immune) into one of the resulting tiles.
Because each resulting tile has a well-defined boundary and
each cell in the biopsy belongs to exactly one tile, a
Hierarchical Strauss model can be fit on each tile to compute
a tile specific value of each parameter in the model. Most
notably, this allows for the computation of a tile-specific
value of <span class="math inline">\(\gamma_{12}\)</span>, which captures the local degree
of tumor cell-immune cell interaction. By modeling this
parameter locally to each tile, we are able to capture not
only the overall level of infiltration in the biopsy, but
also the potentially spatially heterogeneous nature of the
infiltration.</p>
<p>It is worth emphasizing at this point the modular nature of our
pipeline. The clustering method does not depend on the details
of the model used, and the model in turn does not depend on details
of the clustering method. Either can be exchanged for a different
algorithm or model without disrupting the rest of the pipeline,
so long as the output of the clustering algorithm is a spatial
partition of the biopsy. Further, while the value ultimately
selected to summarize the local infiltration at the tile level
will obviously be informed by the exact model selected, there
is considerable latitude in the choice of this quantity as well.
We ultimately decided to summarize local infiltration using
CTIP, defined in section 2.5.</p>
</div>
</div>
<div id="cell-type-interaction-probability" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> Cell Type Interaction Probability<a href="materials-and-methods.html#cell-type-interaction-probability" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="motivation" class="section level3 hasAnchor" number="10.5.1">
<h3><span class="header-section-number">10.5.1</span> Motivation<a href="materials-and-methods.html#motivation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For ease of notation denote the <span class="math inline">\(\gamma_{12}\)</span> parameter
corresponding to tile <span class="math inline">\(k\)</span> of biopsy <span class="math inline">\(i\)</span> by <span class="math inline">\(\gamma_{i_{(k)}}\)</span>. While
<span class="math inline">\(\gamma_{i_{(k)}}\)</span> allows us to distinguish between the relative
abundance of cells of different types and their tendency to
be spatially near one another, as well as the uncertainty
around this tendency, this fundamentally does not capture
the difference between the observed spatial association
and what one would expect to observe by chance given a
particular configuration of tumor cells. In order to properly
assess this, the observed distribution of <span class="math inline">\(\log(\gamma_{i_{(k)}})\)</span>
must be compared to a counterfactual distribution that captures
the tendency when there is no interaction. This motivated
the development of Cell Type Interaction Probability (CTIP).</p>
</div>
<div id="definition" class="section level3 hasAnchor" number="10.5.2">
<h3><span class="header-section-number">10.5.2</span> Definition<a href="materials-and-methods.html#definition" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let <span class="math inline">\(f_{i_{(k)}}(\gamma)\)</span> be the true posterior distribution of <span class="math inline">\(\log(\gamma_{i_{(k)}})\)</span>, and <span class="math inline">\(f_{i_{(k)},0}(\gamma_0)\)</span> be the distribution of the log-interaction parameter absent interaction conditional upon the location of the tumor cells in tile <span class="math inline">\(k\)</span> of biopsy <span class="math inline">\(i\)</span>. Further, assume independence between the true distribution and the null distribution. Then we define the CTIP for tile <span class="math inline">\(k\)</span> of biopsy <span class="math inline">\(i\)</span> by:</p>
<p><span class="math display">\[\begin{equation}
\label{eqn:CTIP}
\text{CTIP} \quad r_{i_{(k)}} = \int \int  I(\gamma &gt; \gamma_0) f_{i_{(k)}}(\gamma) f_{i_{(k)},0}(\gamma_0)  d\gamma d\gamma_0
\tag{5}
\end{equation}\]</span></p>
</div>
<div id="estimating-the-empirical-null-distribution" class="section level3 hasAnchor" number="10.5.3">
<h3><span class="header-section-number">10.5.3</span> Estimating the empirical null distribution<a href="materials-and-methods.html#estimating-the-empirical-null-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The problem of estimating CTIP hinges on being able to estimate and sample from the null distribution of <span class="math inline">\(\log(\gamma_{i_{(k)}})\)</span>, since estimation of the true posterior can proceed as described in section 2.3.
In order to estimate the null distribution
of <span class="math inline">\(\log(\gamma_{i_{(k)}})\)</span> conditional upon the location of the observed tumor cells, we used simulation to construct an empirical “null” distribution.
Recall that when there is no interaction, by definition <span class="math inline">\(\gamma_{i_{(k)}} = 1\)</span>. This means that the Hierarchical Strauss density
reduces to two independent Poisson processes with
intensities <span class="math inline">\(\beta_1, \beta_2\)</span>, which are trivial to simulate.</p>
<p>Given this, the estimation of the null distribution for a given tile proceeded as follows. First, the first order intensity of the lymphocytes was estimated using the standard estimator, <span class="math inline">\(\hat{\beta}_2 = \frac{I_{ik}}{A_{ik}}\)</span> where <span class="math inline">\(I_{ik}\)</span> is the number of immune cells observed in tile <span class="math inline">\(k\)</span> of biopsy <span class="math inline">\(i\)</span> and <span class="math inline">\(A_{ik}\)</span> is the area in <span class="math inline">\(\mu m^2\)</span> of the tile. Next, <span class="math inline">\(S\)</span> simulations of immune cells were generated from a poisson process with intensity <span class="math inline">\(\hat{\beta}_2\)</span>, <span class="math inline">\(S\)</span> being selected a priori. Finally, each simulated set of immune cells was superimposed over the actual observed tumor cells, and samples were drawn from the resulting posterior distribution. These samples, combined across simulations <span class="math inline">\(1...S\)</span> served to estimate the tile specific null distribution of <span class="math inline">\(\log(\gamma)\)</span>.</p>
</div>
<div id="computation-of-ctip" class="section level3 hasAnchor" number="10.5.4">
<h3><span class="header-section-number">10.5.4</span> Computation of CTIP<a href="materials-and-methods.html#computation-of-ctip" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Having established the ability to sample from the true posterior distribution as well as the tile specific null distribution for <span class="math inline">\(\log(\gamma)\)</span>, estimation of CTIP can be done via stochastic integration . Note that equation (<span class="math inline">\(\ref{eqn:CTIP}\)</span>) is equivalent to <span class="math inline">\(E_{\gamma, \gamma_0}[I(\gamma &gt; \gamma_0)]\)</span>. Thus, given <span class="math inline">\(P\)</span> posterior samples<span class="math inline">\(\gamma_1,...,\gamma_P\)</span> from the true posterior <span class="math inline">\(f_\gamma\)</span> and <span class="math inline">\(\gamma_{01},...,\gamma_{0P}\)</span> from the empirical null <span class="math inline">\(f_{\gamma_0}\)</span>, (<span class="math inline">\(\ref{eqn:CTIP}\)</span>) can be estimated by</p>
<p><span class="math display">\[\begin{equation}
\label{eqn:CTIP_EST}
    \widehat{r_{i_{(k)}}} = \frac{1}{P} \sum_{j=1}^P I(\gamma_j &gt; \gamma_{0j})
    \tag{6}
\end{equation}\]</span></p>
</div>
<div id="interpretation-1" class="section level3 hasAnchor" number="10.5.5">
<h3><span class="header-section-number">10.5.5</span> Interpretation<a href="materials-and-methods.html#interpretation-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>By definition, <span class="math inline">\(r_{i_{(k)}} \in [0, 1]\)</span>. As previously mentioned, <span class="math inline">\(r_{i_{(k)}}\)</span> can be
understood as the expected value of an indicator random variable
with respect to the joint distribution of <span class="math inline">\(\gamma_{i_{(k)}}\)</span>
and <span class="math inline">\(\gamma_{i_{(k)},0}\)</span>, this naturally yields
an interpretation of <span class="math inline">\(r_{i_{(k)}}\)</span> as a probability.<br />
Specifically, it can be thought of as measuring the posterior
probability that the observed value of <span class="math inline">\(\gamma_{i_{(k)}}\)</span>
is larger than the value of <span class="math inline">\(\gamma_{i_{(k)},0}\)</span>.<br />
So <span class="math inline">\(r_{i_{(k)}}\)</span> captures immune infiltration for a particular tile, which separately capture the heterogeneity across a biopsy and can be
aggregated across tiles to yield a measure of immune infiltration
on a particular biopsy.</p>
</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-BT00" class="csl-entry">
Baddeley, Adrian, and Rolf Turner. 2000. <span>“Practical Maximum Pseudolikelihood for Spatial Point Patterns: (With Discussion).”</span> <em>Australian &amp; New Zealand Journal of Statistics</em> 42 (3): 283–322.
</div>
<div id="ref-BT05" class="csl-entry">
———. 2005. <span>“Spatstat: An r Package for Analyzing Spatial Point Patterns.”</span> <em>Journal of Statistical Software</em> 12: 1–42.
</div>
<div id="ref-Carpenter17" class="csl-entry">
Carpenter, Bob, Andrew Gelman, Matthew D Hoffman, Daniel Lee, Ben Goodrich, Michael Betancourt, Marcus Brubaker, Jiqiang Guo, Peter Li, and Allen Riddell. 2017. <span>“Stan: A Probabilistic Programming Language.”</span> <em>Journal of Statistical Software</em> 76 (1).
</div>
<div id="ref-King12" class="csl-entry">
King, Ruth, Janine B Illian, Stuart E King, Glenna F Nightingale, and Ditte K Hendrichsen. 2012. <span>“A Bayesian Approach to Fitting Gibbs Processes with Temporal Random Effects.”</span> <em>Journal of Agricultural, Biological, and Environmental Statistics</em> 17 (4): 601–22.
</div>
<div id="ref-Kruschke14" class="csl-entry">
Kruschke, John. 2014. <span>“Doing Bayesian Data Analysis: A Tutorial with r, JAGS, and Stan.”</span>
</div>
<div id="ref-Strauss75" class="csl-entry">
Strauss, David J. 1975. <span>“A Model for Clustering.”</span> <em>Biometrika</em> 62 (2): 467–75.
</div>
<div id="ref-Su15" class="csl-entry">
Su, Yu-Sung, Masanao Yajima, Maintainer Yu-Sung Su, and JAGS SystemRequirements. 2015. <span>“Package <span>‘R2jags’</span>.”</span> <em>R Package Version 0.03-08, URL Http://CRAN. R-Project. Org/Package= R2jags</em>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="simulation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/scratch.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
