<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Results | SPARTYN Supplementary material</title>
  <meta name="description" content="This is “living documentation” of the SPARTYN pipeline" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Results | SPARTYN Supplementary material" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is “living documentation” of the SPARTYN pipeline" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Results | SPARTYN Supplementary material" />
  
  <meta name="twitter:description" content="This is “living documentation” of the SPARTYN pipeline" />
  

<meta name="author" content="Nate Osher" />


<meta name="date" content="2021-11-28" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="materials-and-methods.html"/>
<link rel="next" href="conclusion.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="materials-and-methods.html"><a href="materials-and-methods.html"><i class="fa fa-check"></i><b>3</b> Materials and Methods</a>
<ul>
<li class="chapter" data-level="3.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#overview"><i class="fa fa-check"></i><b>3.1</b> Overview</a></li>
<li class="chapter" data-level="3.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#spatial-point-processes"><i class="fa fa-check"></i><b>3.2</b> Spatial Point Processes</a></li>
<li class="chapter" data-level="3.3" data-path="materials-and-methods.html"><a href="materials-and-methods.html#intensity-thresholding-and-partition"><i class="fa fa-check"></i><b>3.3</b> Intensity Thresholding and Partition</a></li>
<li class="chapter" data-level="3.4" data-path="materials-and-methods.html"><a href="materials-and-methods.html#inference"><i class="fa fa-check"></i><b>3.4</b> Inference</a></li>
<li class="chapter" data-level="3.5" data-path="materials-and-methods.html"><a href="materials-and-methods.html#infiltration-probability"><i class="fa fa-check"></i><b>3.5</b> Infiltration Probability</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="results.html"><a href="results.html"><i class="fa fa-check"></i><b>4</b> Results</a>
<ul>
<li class="chapter" data-level="4.1" data-path="results.html"><a href="results.html#simulation"><i class="fa fa-check"></i><b>4.1</b> Simulation</a></li>
<li class="chapter" data-level="4.2" data-path="results.html"><a href="results.html#application"><i class="fa fa-check"></i><b>4.2</b> Application</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="results.html"><a href="results.html#survival-analysis"><i class="fa fa-check"></i><b>4.2.1</b> Survival Analysis</a></li>
<li class="chapter" data-level="4.2.2" data-path="results.html"><a href="results.html#genomic-associations"><i class="fa fa-check"></i><b>4.2.2</b> Genomic Associations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>5</b> Conclusion</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">SPARTYN Supplementary material</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="results" class="section level1" number="4">
<h1><span class="header-section-number">Chapter 4</span> Results</h1>
<div id="simulation" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Simulation</h2>
<p>To compare our model’s detection of spatial association between
different cell types, we ran a small-scale simulation study in which we
tested the ability of this method to accurately classify positive
interaction across a range of different simulated cell compositions and
spatial associations. For each of four sets of simulations, the number
of simulated tumor cells and lymphocytes were set a priori at <span class="math inline">\(T_s\)</span> and
<span class="math inline">\(L_s\)</span> respectively. Further, interaction was controlled by a parameter
<span class="math inline">\(\phi \in [-1, 1]\)</span>, with <span class="math inline">\(-1\)</span> indicating the most negative possible
interaction and <span class="math inline">\(1\)</span> indicating the most positive possible interaction.</p>
<p>For a given combination of <span class="math inline">\(T_s\)</span> and <span class="math inline">\(L_s\)</span>, the positive and negative
simulations proceeded differently. For the negative simulations (
), <span class="math inline">\(T_s\)</span> tumor cells and <span class="math inline">\(L_s\)</span> lymphocytes were simulated
as independent poisson processes in regions of the window that
overlapped to varying degrees. The overlap was controlled by <span class="math inline">\(\phi\)</span>,
such that the processes overlapped on <span class="math inline">\((100 \cdot (1 + \phi)) \%\)</span> of the
window in which the simulation occurred. Note that when <span class="math inline">\(\phi = -1\)</span>
there was no overlap, and when <span class="math inline">\(\phi = 0\)</span> (complete overlap) the
simulation reduced to simulating two independent Poisson Processes
within the same window.</p>
<p>For the positive simulations (<span class="math inline">\(\phi \in (0, 1]\)</span>), <span class="math inline">\(T_s\)</span> tumor cells were
simulated under a Poisson process. After their locations were
determined, <span class="math inline">\(L_s\)</span> lymphocytes were simulated. For each lymphocyte <span class="math inline">\(l_i\)</span>,
a Bernoulli random variable <span class="math inline">\(C_i \thicksim Bern(\phi)\)</span> was drawn. If
<span class="math inline">\(C_i = 1\)</span>, <span class="math inline">\(l_i\)</span> was simulated within <span class="math inline">\(30\)</span> <span class="math inline">\(\mu m\)</span> of a<br />
randomly selected tumor cell <span class="math inline">\(t_j\)</span>. Otherwise, <span class="math inline">\(l_i\)</span> was simulated from
a Poisson process. Thus, the level of interaction was again controlled
by <span class="math inline">\(p\)</span>, with <span class="math inline">\(p = 0\)</span> now corresponding to two independent Poisson
Processes and <span class="math inline">\(p = 1\)</span> corresponding to a situation in which all
lymphocytes are within <span class="math inline">\(30\)</span> <span class="math inline">\(\mu m\)</span> of at least one tumor cell.</p>
<p>Across the different simulation settings, accurate classification was
possible using IP, with the minimum AUC across simulations being <span class="math inline">\(0.84\)</span>.
For detailed descriptions of simulation settings, simulated data, and
results, see Figure [N].</p>
</div>
<div id="application" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Application</h2>
<p>We applied our method to a data set consisting of 335 images of skin
cutaneous melanoma taken from The Cancer Genome Atlas. Images were
processed as described in section 2.1. Our methods were applied to the
resulting data sets. [GIVE SUMMARY STATISTICS ABOUT TILES?]</p>
<div id="survival-analysis" class="section level3" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Survival Analysis</h3>
<p>In order to assess association between PPPI and survival, we fit a Cox
Proportional Hazards model. In addition to adjusting for average
logit-PPPI, we adjusted for patient level factors such as cancer stage,
age, and sex. In addition, we adjusted for readily calculable tumor
level features, such as number of tumor cells (as a proxy for size) and
logit lymphocyte proportion (the number of TILs divided by the number of
tumor cells and TILs). We found that after adjusting for these other
factors, an increase in logit-PPPI was significantly associated with
increased risk of death (<span class="math inline">\(p &lt; 0.05\)</span>). The same model was fitted with
average logit-PPPI exchanged for the average normalized value of the
estimated Mark Correlation Function evaluated at <span class="math inline">\(r = 30\)</span> (the same as
the radius of interaction used in our model fitting). See Table 1 for
coefficients and standard errors in each model.</p>
</div>
<div id="genomic-associations" class="section level3" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Genomic Associations</h3>
<p>In addition to associations with survival, we investigated the
association between our measurement and gene expression. Gene expression
[RNA-seq] data was acquired for all 335 patients in our sample using
TCGA Assembler [citation here]. Additionally, 42 significantly mutated
genes of interest were identified using previous work investigating the
genomic differences in SCM [TCGA Network, 2015]. Of the 335 patients
in our sample, 240 had gene expression data for all genes of interest,
while 95 were missing data for all genes of interest. We examined the
marginal association between the normalized gene expression data for the
240 patients with complete data and average logit PPPI values. After
correcting for multiple testing using the Benjamini-Hochberg procedure,
we found that the expression of three genes were significantly
associated with average logit PPPI: LRRC37A3, B2M, and TP53. See
supplementary Table [N] for full details on significance of
associations with each gene.</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="materials-and-methods.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="conclusion.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
